<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Evolutionary Transcriptomics • myTAI</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Evolutionary Transcriptomics">
<meta name="description" content="Investigate the evolution of biological processes by capturing evolutionary signatures in transcriptomes (Drost et al. (2018) &lt;doi:10.1093/bioinformatics/btx835&gt;). This package aims to provide a transcriptome analysis environment to quantify the average evolutionary age of genes contributing to a transcriptome of interest.">
<meta property="og:description" content="Investigate the evolution of biological processes by capturing evolutionary signatures in transcriptomes (Drost et al. (2018) &lt;doi:10.1093/bioinformatics/btx835&gt;). This package aims to provide a transcriptome analysis environment to quantify the average evolutionary age of genes contributing to a transcriptome of interest.">
<meta property="og:image" content="https://drostlab.github.io/myTAI/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/myTAI.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Tutorials</h6></li>
    <li><a class="dropdown-item" href="articles/Introduction.html">Introduction</a></li>
    <li><a class="dropdown-item" href="articles/Intermediate.html">Intermediate</a></li>
    <li><a class="dropdown-item" href="articles/Advanced.html">Advanced</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Discussion</h6></li>
    <li><a class="dropdown-item" href="articles/Phylostratigraphy.html">Genomic Phylostratigraphy</a></li>
    <li><a class="dropdown-item" href="articles/Enrichment.html">Enrichment Analyses</a></li>
    <li><a class="dropdown-item" href="articles/Expression.html">Gene Expression Analysis</a></li>
    <li><a class="dropdown-item" href="articles/Taxonomy.html">Taxonomic Retrieval</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="mytai-">myTAI <a class="anchor" aria-label="anchor" href="#mytai-"></a>
</h1>
</div>
<p><a href="https://academic.oup.com/bioinformatics/article/34/9/1589/4772684" class="external-link"><img src="https://img.shields.io/badge/Published%20in-Bioinformatics-126888.svg" alt="Paper link"></a> <a href="https://bioconda.github.io/" class="external-link"><img src="https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat" alt="install with bioconda"></a> <img src="https://api.visitorbadge.io/api/visitors?path=https%3A%2F%2Fgithub.com%2Fdrostlab%2FmyTAI&amp;label=VISITORS&amp;countColor=%23263759&amp;style=flat" alt="Visitors"></p>
</div>
<div class="section level1">
<h1 id="evolutionary-transcriptomics-with-r">Evolutionary Transcriptomics with R<a class="anchor" aria-label="anchor" href="#evolutionary-transcriptomics-with-r"></a>
</h1>
<div class="section level3">
<h3 id="motivation">Motivation<a class="anchor" aria-label="anchor" href="#motivation"></a>
</h3>
<p>In the past years, a variety of studies aimed to uncover the molecular basis of morphological innovation and variation from the evolutionary developmental perspective. These studies often rely on transcriptomic data to establish the molecular patterns driving the complex biological processes underlying phenotypic plasticity.</p>
<p>Although transcriptome information is a useful start to study the molecular mechanisms underlying a biological process of interest (molecular phenotype), they rarely capture <strong>how</strong> these expression patterns emerged in the first place or to <strong>what extent</strong> they are possibly constrained, thereby neglecting the evolutionary history and developmental constraints of genes contributing to the overall pool of expressed transcripts.</p>
<p>To overcome this limitation, the <code>myTAI</code> package introduces procedures summarized under the term <code>evolutionary transcriptomics</code> to integrate gene age information into classical gene expression analysis. <code>Gene age inference</code> can be performed with <a href="https://drostlab.github.io/myTAI/articles/Introduction.html#retrieval-of-phylogenetic-or-taxonomic-information">various existing software</a>, but we recommend using <a href="https://github.com/josuebarrera/GenEra" class="external-link">GenEra</a>, since they address published shortcomings of <code>gene age inference</code> (<a href="https://drostlab.github.io/myTAI/articles/Phylostratigraphy.html">see detailed discussion here</a>). In addition, users can easily retrieve <a href="https://github.com/drostlab/published_phylomaps" class="external-link">previously precomputed gene age information</a> via our data package <a href="https://github.com/LotharukpongJS/phylomapr" class="external-link">phylomapr</a>.</p>
<p>Evolutionary transcriptomics studies can serve as a first approach to screen <em>in silico</em> for the potential existence of evolutionary and developmental constraints within a biological process of interest. This is achieved by quantifying transcriptome conservation patterns and their underlying gene sets in biological processes. The exploratory analysis functions implemented in <code>myTAI</code> provide users with a standardized, automated and statistically sound framework to detect and analyze patterns of evolutionary constraints in any transcriptome dataset of interest.</p>
</div>
<div class="section level3">
<h3 id="detailed-documentation-provided-here">Detailed documentation provided <a href="https://drostlab.github.io/myTAI/articles/Introduction.html">here</a>
<a class="anchor" aria-label="anchor" href="#detailed-documentation-provided-here"></a>
</h3>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>Please cite the following paper when using <code>myTAI</code> for your own research. This will allow me to continue working on this software tool and will motivate me to extend its functionality and usability in the next years. Many thanks in advance!</p>
<blockquote>
<p>Drost et al. <strong>myTAI: evolutionary transcriptomics with R</strong>. <em>Bioinformatics</em> 2018, 34 (9), 1589-1590. <a href="https://academic.oup.com/bioinformatics/advance-article/doi/10.1093/bioinformatics/btx835/4772684" class="external-link">doi:10.1093</a></p>
</blockquote>
<div class="section level3">
<h3 id="dependencies">Dependencies<a class="anchor" aria-label="anchor" href="#dependencies"></a>
</h3>
<p><code>myTAI</code> is using <code>OpenMP</code> to run cpp code in parallel, thus you might need to install additional software.</p>
<p><strong>Mac:</strong></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode bash"><code class="sourceCode bash"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="ex">brew</span> install llvm libomp</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a><span class="bu">cd</span> /usr/local/lib</span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a><span class="fu">ln</span> <span class="at">-s</span> /usr/local/opt/libomp/lib/libomp.dylib ./libomp.dylib</span></code></pre></div>
<p><strong>Linux:</strong> 1. Check if you have an OpenMP-enabled compiler: Open a terminal and run the command <code>gcc --version</code> or <code>g++ --version</code> to check if you have the GNU Compiler Collection (GCC) installed. OpenMP support is typically included in GCC.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode R"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a>If you don<span class="st">'t have GCC installed, you can install it using your distribution'</span>s package manager. For example, on Ubuntu, you can use sudo apt<span class="sc">-</span>get install build<span class="sc">-</span>essential to install GCC.</span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>
<p>Install the libomp library: Open a terminal and run the command appropriate for your package manager:</p>
<p>On Ubuntu or Debian-based systems: <code>sudo apt-get install libomp-dev</code> or <code>sudo apt install libomp-dev</code></p>
<p>On Fedora or CentOS systems: <code>sudo dnf install libomp-devel</code></p>
<p>On Arch Linux: <code>sudo pacman -S libomp</code></p>
</li>
</ol>
<p><strong>Windows:</strong></p>
<p>On Windows, the most common OpenMP implementation is provided by Microsoft Visual C++ (MSVC) compiler, which includes OpenMP support by default. So, if you are using MSVC as your C++ compiler, you should have OpenMP support without any additional installations.</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>Users can install <code>myTAI</code> from <a href="https://CRAN.R-project.org/package=myTAI" class="external-link">CRAN</a>:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install myTAI 0.9.3 from CRAN</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"myTAI"</span>, dependencies <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># install the developer version containing the newest features</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"drostlab/myTAI"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="short-package-description">Short package description<a class="anchor" aria-label="anchor" href="#short-package-description"></a>
</h3>
<p>Using <code>myTAI</code>, any existing or newly generated transcriptome dataset can be combined with evolutionary information (find <a href="https://drostlab.github.io/myTAI/articles/Introduction.html#retrieval-of-phylogenetic-or-taxonomic-information">details here</a>) to retrieve novel insights about the evolutionary conservation of the transcriptome at hand.</p>
<p>For the purpose of performing large scale evolutionary transcriptomics studies, the <code>myTAI</code> package implements the quantification, statistical assessment, and analytics functionality to allow researchers to study the evolution of biological processes by determining stages or periods of evolutionary conservation or variability in transcriptome data.</p>
<p>We hope that <code>myTAI</code> will become the community standard tool to perform evolutionary transcriptomics studies and we are happy to add required functionality upon request.</p>
</div>
</div>
<div class="section level2">
<h2 id="news">NEWS<a class="anchor" aria-label="anchor" href="#news"></a>
</h2>
<p>The current status of the package as well as a detailed history of the functionality of each version of <code>myTAI</code> can be found in the <a href="https://drostlab.github.io/myTAI/news/index.html">NEWS</a> section.</p>
</div>
<div class="section level2">
<h2 id="tutorials">Tutorials<a class="anchor" aria-label="anchor" href="#tutorials"></a>
</h2>
<p>The following tutorials will provide use cases and detailed explanations of how to quantify transcriptome conservation with <code>myTAI</code> and how to interpret the results generated with this software tool.</p>
<ul>
<li><a href="https://drostlab.github.io/myTAI/articles/Introduction.html">Introduction to the myTAI Package</a></li>
<li><a href="https://drostlab.github.io/myTAI/articles/Intermediate.html">Intermediate Concepts of Phylotranscriptomics</a></li>
<li><a href="https://drostlab.github.io/myTAI/articles/Advanced.html">Advanced Topics of Phylotranscriptomics</a></li>
<li><a href="https://drostlab.github.io/myTAI/articles/Enrichment.html">Perform Age Enrichment Analyses</a></li>
<li><a href="https://drostlab.github.io/myTAI/articles/Expression.html">Gene Expression Analysis with myTAI</a></li>
<li><a href="https://drostlab.github.io/myTAI/articles/Taxonomy.html">Taxonomic Information Retrieval</a></li>
<li><a href="https://drostlab.github.io/myTAI/articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a></li>
</ul>
</div>
<div class="section level2">
<h2 id="getting-started-with-mytai">Getting started with <code>myTAI</code>
<a class="anchor" aria-label="anchor" href="#getting-started-with-mytai"></a>
</h2>
<p>Users can also read the tutorials within (<a href="https://posit.co/" class="external-link">RStudio</a>) :</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># source the myTAI package</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/drostlab/myTAI" class="external-link">myTAI</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># look for all tutorials (vignettes) available in the myTAI package</span></span>
<span><span class="co"># this will open your web browser</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/browseVignettes.html" class="external-link">browseVignettes</a></span><span class="op">(</span><span class="st">"myTAI"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># or as single tutorials</span></span>
<span></span>
<span><span class="co"># open tutorial: Introduction to Phylotranscriptomics and myTAI</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"Introduction"</span>, package <span class="op">=</span> <span class="st">"myTAI"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># open tutorial: Intermediate Concepts of Phylotranscriptomics</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"Intermediate"</span>, package <span class="op">=</span> <span class="st">"myTAI"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># open tutorial: Advanced Concepts of Phylotranscriptomics</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"Advanced"</span>, package <span class="op">=</span> <span class="st">"myTAI"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># open tutorial: Age Enrichment Analyses</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"Enrichment"</span>, package <span class="op">=</span> <span class="st">"myTAI"</span><span class="op">)</span></span>
<span> </span>
<span><span class="co"># open tutorial: Gene Expression Analysis with myTAI</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"Expression"</span>, package <span class="op">=</span> <span class="st">"myTAI"</span><span class="op">)</span></span>
<span> </span>
<span> <span class="co"># open tutorial: Taxonomic Information Retrieval with myTAI</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"Taxonomy"</span>, package <span class="op">=</span> <span class="st">"myTAI"</span><span class="op">)</span></span>
<span> </span>
<span> <span class="co"># open tutorial: Discussing Genomic Phylostratigraphy</span></span>
<span> <span class="fu"><a href="https://rdrr.io/r/utils/vignette.html" class="external-link">vignette</a></span><span class="op">(</span><span class="st">"Phylostratigraphy"</span>, package <span class="op">=</span> <span class="st">"myTAI"</span><span class="op">)</span></span></code></pre></div>
<p>In the <code>myTAI</code> framework users can find:</p>
<div class="section level4">
<h4 id="phylotranscriptomics-measures">Phylotranscriptomics Measures:<a class="anchor" aria-label="anchor" href="#phylotranscriptomics-measures"></a>
</h4>
<ul>
<li>
<code><a href="reference/TAI.html">TAI()</a></code> : Function to compute the Transcriptome Age Index (TAI)</li>
<li>
<code><a href="reference/TDI.html">TDI()</a></code> : Function to compute the Transcriptome Divergence Index (TDI)</li>
<li>
<code><a href="reference/TPI.html">TPI()</a></code> : Function to compute the Transcriptome Polymorphism Index (TPI)</li>
<li>
<code><a href="reference/REMatrix.html">REMatrix()</a></code> : Function to compute the relative expression profiles of all phylostrata or divergence-strata</li>
<li>
<code><a href="reference/RE.html">RE()</a></code> : Function to transform mean expression levels to relative expression levels</li>
<li>
<code><a href="reference/pTAI.html">pTAI()</a></code> : Compute the Phylostratum Contribution to the global TAI</li>
<li>
<code><a href="reference/pTDI.html">pTDI()</a></code> : Compute the Divergence Stratum Contribution to the global TDI</li>
<li>
<code><a href="reference/pMatrix.html">pMatrix()</a></code> : Compute Partial TAI or TDI Values</li>
<li>
<code><a href="reference/pStrata.html">pStrata()</a></code> : Compute Partial Strata Values</li>
</ul>
</div>
<div class="section level4">
<h4 id="visualization-and-analytics-tools">Visualization and Analytics Tools:<a class="anchor" aria-label="anchor" href="#visualization-and-analytics-tools"></a>
</h4>
<ul>
<li>
<code><a href="reference/PlotSignature.html">PlotSignature()</a></code> : Main visualization function to plot evolutionary signatures across transcriptomes</li>
<li>
<code><a href="reference/PlotPattern.html">PlotPattern()</a></code> : Base graphics function to plot evolutionary signatures across transcriptomes</li>
<li>
<code><a href="reference/PlotContribution.html">PlotContribution()</a></code> : Plot Cumulative Transcriptome Index</li>
<li>
<code><a href="reference/PlotCorrelation.html">PlotCorrelation()</a></code> : Function to plot the correlation between phylostratum values and divergence-stratum values</li>
<li>
<code><a href="reference/PlotRE.html">PlotRE()</a></code> : Function to plot the relative expression profiles</li>
<li>
<code><a href="reference/PlotBarRE.html">PlotBarRE()</a></code> : Function to plot the mean relative expression levels of phylostratum or divergence-stratum classes as barplot</li>
<li>
<code><a href="reference/PlotMeans.html">PlotMeans()</a></code> : Function to plot the mean expression profiles of age categories</li>
<li>
<code><a href="reference/PlotMedians.html">PlotMedians()</a></code> : Function to plot the median expression profiles of age categories</li>
<li>
<code><a href="reference/PlotVars.html">PlotVars()</a></code> : Function to plot the expression variance profiles of age categories</li>
<li>
<code><a href="reference/PlotDistribution.html">PlotDistribution()</a></code> : Function to plot the frequency distribution of genes within the corresponding age categories</li>
<li>
<code><a href="reference/PlotCategoryExpr.html">PlotCategoryExpr()</a></code> : Plot the Expression Levels of each Age or Divergence Category as Barplot or Violinplot</li>
<li>
<code><a href="reference/PlotEnrichment.html">PlotEnrichment()</a></code> : Plot the Phylostratum or Divergence Stratum Enrichment of a given Gene Set</li>
<li>
<code><a href="https://rdrr.io/pkg/myTAI/man/PlotGeneSet.html" class="external-link">PlotGeneSet()</a></code> : Plot the Expression Profiles of a Gene Set</li>
<li>
<code><a href="reference/PlotGroupDiffs.html">PlotGroupDiffs()</a></code> : Plot the significant differences between gene expression distributions of PS or DS groups</li>
<li>
<code><a href="reference/PlotSelectedAgeDistr.html">PlotSelectedAgeDistr()</a></code> : Plot the PS or DS distribution of a selected set of genes</li>
</ul>
</div>
<div class="section level4">
<h4 id="a-statistical-framework-and-test-statistics">A Statistical Framework and Test Statistics:<a class="anchor" aria-label="anchor" href="#a-statistical-framework-and-test-statistics"></a>
</h4>
<ul>
<li>
<code><a href="reference/FlatLineTest.html">FlatLineTest()</a></code> : Function to perform the <strong>Flat Line Test</strong> that quantifies the statistical significance of an observed phylotranscriptomics pattern (significant deviation from a frat line = evolutionary signal)</li>
<li>
<code><a href="reference/ReductiveHourglassTest.html">ReductiveHourglassTest()</a></code> : Function to perform the <strong>Reductive Hourglass Test</strong> that statistically evaluates the existence of a phylotranscriptomic hourglass pattern (hourglass model)</li>
<li>
<code><a href="reference/EarlyConservationTest.html">EarlyConservationTest()</a></code> : Function to perform the <strong>Reductive Early Conservation Test</strong> that statistically evaluates the existence of a monotonically increasing phylotranscriptomic pattern (early conservation model)</li>
<li>
<code><a href="reference/LateConservationTest.html">LateConservationTest()</a></code> : Function to perform the <strong>Reductive Late Conservation Test</strong> that statistically evaluates the existence of a monotonically decreasing phylotranscriptomic pattern (early divergence model)</li>
<li>
<code><a href="reference/ReverseHourglassTest.html">ReverseHourglassTest()</a></code>: Function to perform the <strong>Reverse Hourglass Test</strong> that statistically evaluates the existence of a reverse hourglass pattern (low-high-low)</li>
<li>
<code><a href="reference/PairwiseTest.html">PairwiseTest()</a></code>: Function to perform the <strong>Pairwise Difference Test</strong> that statistically evaluates the difference in TAI (or TDI) between two constrasts, e.g. male vs female samples.</li>
<li>
<code><a href="reference/EnrichmentTest.html">EnrichmentTest()</a></code> : Phylostratum or Divergence Stratum Enrichment of a given Gene Set based on Fisher’s Test</li>
<li>
<code><a href="reference/bootMatrix.html">bootMatrix()</a></code> : Compute a Permutation Matrix for Test Statistics</li>
</ul>
<p>All functions also include visual analytics tools to quantify the goodness of test statistics.</p>
</div>
<div class="section level4">
<h4 id="differential-gene-expression-analysis">Differential Gene Expression Analysis<a class="anchor" aria-label="anchor" href="#differential-gene-expression-analysis"></a>
</h4>
<ul>
<li>
<code><a href="reference/DiffGenes.html">DiffGenes()</a></code> : Implements Popular Methods for Differential Gene Expression Analysis</li>
<li>
<code><a href="reference/CollapseReplicates.html">CollapseReplicates()</a></code> : Combine Replicates in an ExpressionSet</li>
<li>
<code><a href="reference/CombinatorialSignificance.html">CombinatorialSignificance()</a></code> : Compute the Statistical Significance of Each Replicate Combination</li>
<li>
<code><a href="reference/Expressed.html">Expressed()</a></code> : Filter Expression Levels in Gene Expression Matrices (define expressed genes)</li>
<li>
<code><a href="reference/SelectGeneSet.html">SelectGeneSet()</a></code> : Select a Subset of Genes in an ExpressionSet</li>
<li>
<code><a href="reference/PlotReplicateQuality.html">PlotReplicateQuality()</a></code> : Plot the Quality of Biological Replicates</li>
<li>
<code><a href="reference/GroupDiffs.html">GroupDiffs()</a></code> : Quantify the significant differences between gene expression distributions of PS or DS groups</li>
</ul>
</div>
<div class="section level4">
<h4 id="taxonomic-information-retrieval">Taxonomic Information Retrieval<a class="anchor" aria-label="anchor" href="#taxonomic-information-retrieval"></a>
</h4>
<ul>
<li>
<code><a href="https://rdrr.io/pkg/myTAI/man/taxonomy.html" class="external-link">taxonomy()</a></code> : Retrieve Taxonomic Information for any Organism of Interest</li>
</ul>
</div>
<div class="section level4">
<h4 id="minor-functions-for-better-usibility-and-additional-analyses">Minor Functions for Better Usibility and Additional Analyses<a class="anchor" aria-label="anchor" href="#minor-functions-for-better-usibility-and-additional-analyses"></a>
</h4>
<ul>
<li>
<code><a href="reference/MatchMap.html">MatchMap()</a></code> : Match a Phylostratigraphic Map or Divergence Map with a ExpressionMatrix</li>
<li>
<code><a href="reference/tf.html">tf()</a></code> : Transform Gene Expression Levels</li>
<li>
<code><a href="reference/tfPS.html">tfPS()</a></code> : Transform Phylostratum Levels</li>
<li>
<code><a href="reference/tfStability.html">tfStability()</a></code> : Perform Permutation Tests Under Different Transformations</li>
<li>
<code><a href="reference/age.apply.html">age.apply()</a></code> : Age Category Specific apply Function</li>
<li>
<code><a href="reference/ecScore.html">ecScore()</a></code> : Compute the Hourglass Score for the EarlyConservationTest</li>
<li>
<code><a href="reference/lcScore.html">lcScore()</a></code> : Compute the Hourglass Score for the LateConservationTest</li>
<li>
<code><a href="reference/pairScore.html">pairScore()</a></code> : Compute the PairwiseTest Score for the PairwiseTest</li>
<li>
<code><a href="reference/geom.mean.html">geom.mean()</a></code> : Geometric Mean</li>
<li>
<code><a href="reference/harm.mean.html">harm.mean()</a></code> : Harmonic Mean</li>
<li>
<code><a href="reference/omitMatrix.html">omitMatrix()</a></code> : Compute TAI or TDI Profiles Omitting a Given Gene</li>
<li>
<code><a href="reference/rhScore.html">rhScore()</a></code> : Compute the Hourglass Score for the Reductive Hourglass Test</li>
<li>
<code><a href="reference/reversehourglassScore.html">reversehourglassScore()</a></code>: Compute the Reverse Hourglass Score for the Reverse Hourglass Test</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="studies-that-successfully-used-mytai-to-quantify-transcriptome-conservation">Studies that successfully used <code>myTAI</code> to quantify transcriptome conservation:<a class="anchor" aria-label="anchor" href="#studies-that-successfully-used-mytai-to-quantify-transcriptome-conservation"></a>
</h2>
<blockquote>
<ul>
<li><p><em>A transcriptomic hourglass in brown algae</em> JS Lotharukpong, M Zheng, R Luthringer, H-G Drost, SM Coelho - <strong>Nature</strong>, 2024</p></li>
<li><p><em>Genome assemblies of 11 bamboo species highlight diversification induced by dynamic subgenome dominance</em> PF Ma, YL Liu, C Guo, G Jin, ZH Guo, L Mao, YZ Yang… - <strong>Nature Genetics</strong>, 2024</p></li>
<li><p><em>Hemichordate cis-regulatory genomics and the gene expression dynamics of deuterostomes</em> A Pérez-Posada, CY Lin, TP Fan, CY Lin, YC Chen… - <strong>Nature Ecology &amp; Evolution</strong>, 2024</p></li>
<li><p><em>Comparison between 16S rRNA and shotgun sequencing in colorectal cancer, advanced colorectal lesions, and healthy human gut microbiota</em> D Bars-Cortina, E Ramon, B Rius-Sansalvador… - <strong>BMC genomics</strong>, 2024</p></li>
<li><p><em>Heat stress reprograms herbivory-induced defense responses in potato plants</em> J Zhong, J Zhang, Y Zhang, Y Ge, W He, C Liang… - <strong>BMC Plant Biology</strong>, 2024</p></li>
<li><p><em>The transcriptomic signature of adaptations associated with perfume collection in orchid bees</em> K Darragh, SR Ramírez - <strong>Journal of Evolutionary Biology</strong>, 2024</p></li>
<li><p><em>Proteomic dynamics revealed sex‐biased responses to combined heat‐drought stress in Marchantia</em> S Guerrero, V Roces, L García‐Campa, L Valledor… - <strong>Journal of Integrative Plant Biology</strong>, 2024</p></li>
<li><p><em>Evolution of gene networks underlying adaptation to drought stress in the wild tomato Solanum chilense</em> K Wei, S Sharifova, X Zhao, N Sinha, H Nakayama… - <strong>Molecular Ecology</strong>, 2024</p></li>
<li><p><em>Conserved and specific gene expression patterns in the embryonic development of tardigrades</em> C Li, Z Yang, X Xu, L Meng, S Liu, D Yang - <strong>Evolution &amp; Development</strong>, 2024</p></li>
<li><p><em>The functions and factors governing fungal communities and diversity in agricultural waters: insights into the ecosystem services aquatic mycobiota provide</em> P Pham, Y Shi, I Khan, M Sumarah, J Renaud… - <strong>Frontiers in Microbiology</strong>, 2024</p></li>
<li><p><em>An evolutionary timeline of the oxytocin signaling pathway</em> AM Sartorius, J Rokicki, S Birkeland, F Bettella, C Barth… - <strong>Communications Biology</strong>, 2024</p></li>
<li><p><em>The Evolution of Foraging Webs is Associated with Young Genes in Araneoidea Spiders</em> A Jia, T Yang, W Hu, S Ma, Z Zhang, Y Wang - <strong>Available at SSRN 4383994</strong></p></li>
<li><p><em>Multiplexed transcriptomic analyses of the plant embryonic hourglass</em> H Wu, R Zhang, KJ Niklas, MJ Scanlon - <strong>BioRxiv</strong>, 2024</p></li>
<li><p><em>Somatic embryogenesis of grapevine (Vitis vinifera) expresses a transcriptomic hourglass</em> S Koska, D Leljak-Levanic, N Malenica, K Bigovic Villi… - <strong>BioRxiv</strong>, 2024</p></li>
<li><p><em>Brachiopod genome unveils the evolution of the BMP–Chordin network in bilaterian body patterning</em> TD Lewin, K Shimizu, IJY Liao, ME Chen, K Endo… - <strong>BioRxiv</strong>, 2024</p></li>
<li><p><em>The angiosperm seed life cycle follows a developmental reverse hourglass</em> AA Sami, L Bentsink, MAS Artur - <strong>BioRxiv</strong>, 2024</p></li>
<li><p><em>Evolutionary trends in the emergence of skeletal cell types</em> A Damatac, S Koska, KK Ullrich, T Domazet-Loso… - <strong>BioRxiv</strong>, 2024</p></li>
<li><p><em>Transcriptome age of individual cell types in Caenorhabditis elegans</em> F Ma, C Zheng - <strong>Proceedings of the National Academy of Sciences</strong>, 2023</p></li>
<li><p><em>Single-cell atlases of two lophotrochozoan larvae highlight their complex evolutionary histories</em> L Piovani, DJ Leite, LA Yañez Guerra, F Simpson… - <strong>Science Advances</strong>, 2023</p></li>
<li><p><em>oggmap: a Python package to extract gene ages per orthogroup and link them with single-cell RNA data</em> KK Ullrich, NE Glytnasi - <strong>Bioinformatics</strong>, 2023</p></li>
<li><p><em>Discovery of putative long non-coding RNAs expressed in the eyes of Astyanax mexicanus (Actinopterygii: Characidae)</em> I Batista da Silva, D Aciole Barbosa, KF Kavalco… - <strong>Scientific Reports</strong>, 2023</p></li>
<li><p><em>An ancient split of germline and somatic stem cell lineages in Hydra</em> C Nishimiya-Fujisawa, H Petersen, TC-T Koubková-Yu, C Noda, S Shigenobu, J Bageritz, T Fujisawa, O Simakov, S Kobayashi, TW Holstein - <strong>BioRxiv</strong>, 2023</p></li>
<li><p><em>Oxytocin receptor expression patterns in the human brain across development</em> J Rokicki, T Kaufmann, A-MG de Lange, D van der Meer, S Bahrami, AM Sartorius, UK Haukvik, NE Steen, E Schwarz, DJ Stein, T Nærland, OA Andreassen, LT Westlye, DS Quintana - <strong>Neuropsychopharmacology</strong>, 2022</p></li>
<li><p><em>The Phylotranscriptomic Hourglass Pattern in Fungi: An Updated Model</em> Y Xie, HS Kwan, PL Chan, WJ Wu, J Chiou, J Chang <strong>BioRxiv</strong>, 2022</p></li>
<li><p><em>Embryo-Like Features in Developing Bacillus subtilis Biofilms</em> M Futo, L Opašić, S Koska, N Čorak, T Široki, V Ravikumar, A Thorsell, M Lenuzzi, D Kifer, M Domazet-Lošo, K Vlahoviček, I Mijakovic, T Domazet-Lošo - <strong>Molecular Biology and Evolution</strong>, 2021</p></li>
<li><p><em>New Genes Interacted With Recent Whole-Genome Duplicates in the Fast Stem Growth of Bamboos</em> G Jin, P-F Ma, X Wu, L Gu, M Long, C Zhang, DZ Li - <strong>Molecular Biology and Evolution</strong>, 2021</p></li>
<li><p><em>Evolutionary transcriptomics of metazoan biphasic life cycle supports a single intercalation origin of metazoan larvae</em> J Wang, L Zhang, S Lian, Z Qin, X Zhu, X Dai, Z Huang et al. - <strong>Nature Ecology &amp; Evolution</strong>, 2020</p></li>
<li><p><em>Pervasive convergent evolution and extreme phenotypes define chaperone requirements of protein homeostasis</em> Y Draceni, S Pechmann - <strong>Proceedings of the National Academy of Sciences</strong>, 2019</p></li>
<li><p><em>Reconstructing the transcriptional ontogeny of maize and sorghum supports an inverse hourglass model of inflorescence development</em> S Leiboff, S Hake - <strong>Current Biology</strong>, 2019</p></li>
<li><p><em>The Transcriptional Landscape of Polyploid Wheats and their Diploid Ancestors during Embryogenesis and Grain Development</em> D Xiang, TD Quilichini, Z Liu, P Gao, Y Pan et al. - <strong>The Plant Cell</strong>, 2019</p></li>
<li><p><em>Pervasive convergent evolution and extreme phenotypes define chaperone requirements of protein homeostasis</em> Y Draceni, S Pechmann - <strong>Proceedings of the National Academy of Sciences</strong>, 2019</p></li>
<li><p><em>A unicellular relative of animals generates a layer of polarized cells by actomyosin-dependent cellularization</em> O Dudin, A Ondracka, X Grau-Bové, AAB Haraldsen et al. - <strong>eLife</strong>, 2019</p></li>
<li><p><em>Gene Expression Does Not Support the Developmental Hourglass Model in Three Animals with Spiralian Development</em> L Wu, KE Ferger, JD Lambert - <strong>Molecular Biology and Evolution</strong>, 2019</p></li>
<li><p><em>Phylostratr: a framework for phylostratigraphy</em> Z Arendsee, J Li, U Singh, A Seetharam et al. - <strong>Bioinformatics</strong>, 2019</p></li>
<li><p><em>Algorithms for synteny-based phylostratigraphy and gene origin classification</em> Z Arendsee - 2019</p></li>
<li><p><em>Elucidating the endogenous synovial fluid proteome and peptidome of inflammatory arthritis using label-free mass spectrometry</em> SM Mahendran, EC Keystone, RJ Krawetz et al. - <strong>Clinical proteomics</strong>, 2019</p></li>
<li><p><em>Environmental DNA reveals landscape mosaic of wetland plant communities</em> ME Shackleton, GN Rees, G Watson et al. - <strong>Global Ecology and Conservation</strong>, 2019</p></li>
<li><p><em>Developmental constraints on genome evolution in four bilaterian model species</em> J Liu, M Robinson-Rechavi - <strong>Genome Biology and Evolution</strong>, 2018</p></li>
<li><p><em>Mapping selection within Drosophila melanogaster embryo’s anatomy</em> I Salvador-Martínez et al. - <strong>Molecular Biology and Evolution</strong>, 2017</p></li>
<li><p><em>Distribution and diversity of enzymes for polysaccharide degradation in fungi</em> R Berlemont - <strong>Scientific reports</strong>, 2017</p></li>
<li><p><em>The origins and evolutionary history of human non-coding RNA regulatory networks</em> M Sherafatian, SJ Mowla - <strong>Journal of bioinformatics and computational biology</strong>, 2017</p></li>
<li><p><em>High expression of new genes in trochophore enlightening the ontogeny and evolution of trochozoans</em> F Xu, T Domazet-Lošo, D Fan, TL Dunwell, L Li et al. - <strong>Scientific reports</strong>, 2016</p></li>
<li><p><em>Evidence for active maintenance of phylotranscriptomic hourglass patterns in animal and plant embryogenesis</em> HG Drost, A Gabel, I Grosse, M Quint - <strong>Molecular Biology and Evolution</strong>, 2015</p></li>
</ul>
</blockquote>
</div>
<div class="section level2">
<h2 id="discussions-and-bug-reports">Discussions and Bug Reports<a class="anchor" aria-label="anchor" href="#discussions-and-bug-reports"></a>
</h2>
<p>I would be very happy to learn more about potential improvements of the concepts and functions provided in this package.</p>
<p>Furthermore, in case you find some bugs or need additional (more flexible) functionality of parts of this package, please let me know:</p>
<p><a href="https://github.com/drostlab/myTAI/issues" class="external-link uri">https://github.com/drostlab/myTAI/issues</a></p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Domazet-Lošo T. and Tautz D. <strong>A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns</strong>. <em>Nature</em> (2010) 468: 815-8.</p>
<p>Quint M, Drost HG, et al. <strong>A transcriptomic hourglass in plant embryogenesis</strong>. <em>Nature</em> (2012) 490: 98-101.</p>
<p>Drost HG, Gabel A, Grosse I, Quint M. <strong>Evidence for Active Maintenance of Phylotranscriptomic Hourglass Patterns in Animal and Plant Embryogenesis</strong>. <em>Mol. Biol. Evol.</em> (2015) 32 (5): 1221-1231.</p>
<p>Drost HG, Bellstädt J, Ó’Maoiléidigh DS, Silva AT, Gabel A, Weinholdt C, Ryan PT, Dekkers BJW, Bentsink L, Hilhorst H, Ligterink W, Wellmer F, Grosse I, and Quint M. <strong>Post-embryonic hourglass patterns mark ontogenetic transitions in plant development</strong>. <em>Mol. Biol. Evol.</em> (2016) <a href="https://academic.oup.com/mbe/article/33/5/1158/2580081" class="external-link">doi:10.1093/molbev/msw039</a></p>
</div>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://cloud.r-project.org/package=myTAI" class="external-link">View on CRAN</a></li>
<li><a href="https://github.com/drostlab/myTAI/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/drostlab/myTAI/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="https://www.r-project.org/Licenses/GPL-2" class="external-link">GPL-2</a></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing myTAI</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Hajk-Georg Drost <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-1567-306X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Jaruwatana Sodai Lotharukpong <br><small class="roles"> Author, contributor </small> <a href="https://orcid.org/0000-0002-3475-0980" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Nikola Kalábová <br><small class="roles"> Author, contributor </small> <a href="https://orcid.org/0009-0007-6384-8809" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Filipa Martins Costa <br><small class="roles"> Author, contributor </small>  </li>
<li>Kristian K Ullrich <br><small class="roles"> Author, contributor </small> <a href="https://orcid.org/0000-0003-4308-9626" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Stefan Manolache <br><small class="roles"> Author, contributor </small> <a href="https://orcid.org/0009-0006-3326-985X" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich, Stefan Manolache.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
