<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="Detect differentially expressed genes (DEGs) in a standard ExpressionSet object."><title>Differential Gene Expression Analysis — DiffGenes • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Differential Gene Expression Analysis — DiffGenes"><meta property="og:description" content="Detect differentially expressed genes (DEGs) in a standard ExpressionSet object."><meta property="og:image" content="https://drostlab.github.io/myTAI/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    <a class="dropdown-item" href="../articles/Enrichment.html">Enrichment Analyses</a>
    <a class="dropdown-item" href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    <a class="dropdown-item" href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    <a class="dropdown-item" href="../articles/Introduction.html">Introduction to the myTAI package</a>
    <a class="dropdown-item" href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Differential Gene Expression Analysis</h1>
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/DiffGenes.R" class="external-link"><code>R/DiffGenes.R</code></a></small>
      <div class="d-none name"><code>DiffGenes.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Detect differentially expressed genes (DEGs) in a standard <code>ExpressionSet</code> object.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">DiffGenes</span><span class="op">(</span></span>
<span>  <span class="va">ExpressionSet</span>,</span>
<span>  <span class="va">nrep</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"foldchange"</span>,</span>
<span>  p.adjust.method <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  comparison <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  alpha <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  filter.method <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  n <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  stage.names <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>ExpressionSet</dt>
<dd><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></dd>


<dt>nrep</dt>
<dd><p>either a numeric value specifying the constant number of replicates per stage or a numeric vector specifying the variable number of replicates for each stage position.</p></dd>


<dt>method</dt>
<dd><p>method to detect differentially expressed genes.</p></dd>


<dt>p.adjust.method</dt>
<dd><p>p value correction method that is passed to <code><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></code>.
Available options are:</p><ul><li><p><code>p.adjust.method = "BH"</code> (Benjamini-Hochberg correction)</p></li>
<li><p><code>p.adjust.method = "bonferroni"</code> (Bonferroni correction)</p></li>
<li><p><code>p.adjust.method = "holm"</code></p></li>
<li><p><code>p.adjust.method = "hochberg"</code></p></li>
<li><p><code>p.adjust.method = "hommel"</code></p></li>
<li><p><code>p.adjust.method = "BY"</code></p></li>
<li><p><code>p.adjust.method = "fdr"</code></p></li>
</ul><p>If <code>p.adjust.method = NULL</code> (Default) then no p-value correction is performed.</p></dd>


<dt>comparison</dt>
<dd><p>a character string specifying whether genes having fold-change or p-values
 below, above, or below AND above (both) the <code>alpha</code> value should be excluded from the dataset.
 In case <code>comparison = "both"</code> is chosen, the <code>cut.off</code> argument must be a two dimensional vector defining the lower <code>alpha</code> value at the first position and the upper <code>alpha</code> value
at the second position.</p></dd>


<dt>alpha</dt>
<dd><p>a numeric value specifying the cut-off value above which Genes fulfilling the corresponding fold-change, log-fold-change, or p-value should be retained and returned by <code>DiffGenes</code>.</p></dd>


<dt>filter.method</dt>
<dd><p>a method how to <code>alpha</code> values in multiple stages. Options are <code>"const"</code>, <code>"min-set"</code>, and <code>"n-set"</code>.</p></dd>


<dt>n</dt>
<dd><p>a numeric value for <code>method = "n-set"</code>.</p></dd>


<dt>stage.names</dt>
<dd><p>a character vector specifying the new names of collapsed stages.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>All methods to perform detection of differentially expressed genes assume that your input
dataset has been normalized before passing it to <em>DiffGenes</em>. For RNA-Seq data
<em>DiffGenes</em> assumes that the libraries have been normalized to have the same size, i.e.,
to have the same expected column sum under the null hypothesis.</p>
<p>Available methods for the detection of differentially expressed genes:</p>
<ul><li><p><code>method = "foldchange"</code>: ratio of replicate geometric means between developmental stages.
 Here, the <em>DiffGenes</em> functions assumes that absolute expression levels are stored in your input <code>ExpresisonSet</code>.</p></li>
<li><p><code>method = "log-foldchange"</code>: difference of replicate arithmetic means between developmental stages. Here, the <em>DiffGenes</em> functions assumes that <em>log2a</em>
transformed expression levels are stored in your input <code>ExpresisonSet</code>.</p></li>
<li><p><code>method = "t.test"</code>: Welch t.test between replicate expression levels of two samples.</p></li>
<li><p><code>method = "wilcox.test"</code>: Wilcoxon Rank Sum Test between replicate expression levels of two samples.</p></li>
</ul><p>Exclude non differentially expressed genes from the result dataset:</p>
<p>When specifying the <code>alpha</code> argument you furthermore, need to specify the <code>filter.method</code> to decide how non differentially expressed genes should be classified in multiple sample comparisons and which genes should be retained in the final dataset returned by <code>DiffGenes</code>. In other words, all genes &lt; <code>alpha</code> based on the following <code>filter.method</code> are removed from the result dataset.</p>
<p>Following extraction criteria are implemented in this function:</p>
<ul><li><p><code>const</code>: all genes that have at least one sample comparison that undercuts or exceeds the <code>alpha</code> value <code>cut.off</code> will be excluded from the <code>ExpressionSet</code>. Hence, for a 7 stage <code>ExpressionSet</code> genes passing the <code>alpha</code> threshold in 6 stages will be retained in the <code>ExpressionSet</code>.</p></li>
<li><p><code>min-set</code>: genes passing the <code>alpha</code> value in <code>ceiling(n/2)</code> stages will be retained in the <code>ExpressionSet</code>, where <em>n</em> is the number of stages in the <code>ExpressionSet</code>.</p></li>
<li><p><code>n-set</code>: genes passing the <code>alpha</code> value in <code>n</code> stages will be retained in the <code>ExpressionSet</code>. Here, the argument <code>n</code> needs to be specified.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>In case input <code>ExpressionSet</code> objects store 0 values, internally all expression levels are 
shifted by <code>+1</code> to allow sufficient fold-change and p-value computations. Additionally, a warning
is printed to the console in case expression levels have been automatically shifted.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="Expressed.html">Expressed</a></code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Detection of DEGs using the fold-change measure</span></span></span>
<span class="r-in"><span><span class="va">DEGs</span> <span class="op">&lt;-</span> <span class="fu">DiffGenes</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span> ,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>                  nrep          <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>                  comparison    <span class="op">=</span> <span class="st">"below"</span>,</span></span>
<span class="r-in"><span>                  method        <span class="op">=</span> <span class="st">"foldchange"</span>,</span></span>
<span class="r-in"><span>                  stage.names   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S1"</span>,<span class="st">"S2"</span>,<span class="st">"S3"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">DEGs</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Phylostratum      GeneID    S1-&gt;S2    S1-&gt;S3    S2-&gt;S1    S2-&gt;S3    S3-&gt;S1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1 at1g01040.2 1.6706342 2.0767465 0.5985751 1.2430887 0.4815224</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1 at1g01050.1 1.0231405 1.2788699 0.9773829 1.2499456 0.7819403</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            1 at1g01070.1 1.3087104 1.4044710 0.7641110 1.0731717 0.7120119</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            1 at1g01080.2 0.7786521 0.7286129 1.2842706 0.9357362 1.3724709</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            1 at1g01090.1 0.3744682 0.2255944 2.6704536 0.6024393 4.4327345</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6            1 at1g01120.1 0.9110850 0.9000931 1.0975925 0.9879354 1.1109962</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      S3-&gt;S2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 0.8044478</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 0.8000348</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 0.9318173</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4 1.0686773</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 1.6599182</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6 1.0122119</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Detection of DEGs using the log-fold-change measure</span></span></span>
<span class="r-in"><span><span class="co"># when choosing method = "log-foldchange" it is assumed that</span></span></span>
<span class="r-in"><span><span class="co"># your input expression matrix stores log2 expression levels </span></span></span>
<span class="r-in"><span><span class="va">log.DEGs</span> <span class="op">&lt;-</span> <span class="fu">DiffGenes</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="fu"><a href="tf.html">tf</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,<span class="va">log2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                      nrep          <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>                      comparison    <span class="op">=</span> <span class="st">"below"</span>,</span></span>
<span class="r-in"><span>                      method        <span class="op">=</span> <span class="st">"log-foldchange"</span>,</span></span>
<span class="r-in"><span>                      stage.names   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S1"</span>,<span class="st">"S2"</span>,<span class="st">"S3"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">log.DEGs</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Phylostratum      GeneID      S1-&gt;S2     S1-&gt;S3      S2-&gt;S1      S2-&gt;S3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1 at1g01040.2  0.74039587  1.0543251 -0.74039587  0.31392926</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1 at1g01050.1  0.03300429  0.3548696 -0.03300429  0.32186526</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            1 at1g01070.1  0.38814592  0.4900268 -0.38814592  0.10188092</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            1 at1g01080.2 -0.36094927 -0.4567755  0.36094927 -0.09582626</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            1 at1g01090.1 -1.41708484 -2.1481970  1.41708484 -0.73111211</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>       S3-&gt;S1      S3-&gt;S2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1 -1.0543251 -0.31392926</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2 -0.3548696 -0.32186526</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3 -0.4900268 -0.10188092</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  0.4567755  0.09582626</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5  2.1481970  0.73111211</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Remove fold-change values &lt; 2 from the dataset:</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## first have a look at the range of fold-change values of all genes </span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">DEGs</span><span class="op">[</span> , <span class="fl">3</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,<span class="fl">2</span>,<span class="va">range</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>          S1-&gt;S2      S1-&gt;S3     S2-&gt;S1      S2-&gt;S3     S3-&gt;S1      S3-&gt;S2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1,]  0.1081031  0.04752967 0.06022388  0.06115996  0.0244683  0.06450549</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2,] 16.6047090 40.86920354 9.25042630 15.50255551 21.0394884 16.35056753</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># now remove genes undercutting the alpha = 2 threshold</span></span></span>
<span class="r-in"><span><span class="co"># hence, remove genes having p-values &lt;= 0.05 in at</span></span></span>
<span class="r-in"><span><span class="co"># least one sample comparison</span></span></span>
<span class="r-in"><span><span class="va">DEGs.alpha</span> <span class="op">&lt;-</span> <span class="fu">DiffGenes</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">250</span> ,<span class="fl">1</span><span class="op">:</span><span class="fl">8</span><span class="op">]</span>,</span></span>
<span class="r-in"><span>                        nrep          <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>                        method        <span class="op">=</span> <span class="st">"t.test"</span>,</span></span>
<span class="r-in"><span>                        alpha         <span class="op">=</span> <span class="fl">0.05</span>,</span></span>
<span class="r-in"><span>                        comparison    <span class="op">=</span> <span class="st">"above"</span>,</span></span>
<span class="r-in"><span>                        filter.method <span class="op">=</span> <span class="st">"n-set"</span>,</span></span>
<span class="r-in"><span>                        n             <span class="op">=</span> <span class="fl">1</span>,</span></span>
<span class="r-in"><span>                        stage.names   <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"S1"</span>,<span class="st">"S2"</span>,<span class="st">"S3"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># now again have a look at the range and find</span></span></span>
<span class="r-in"><span><span class="co"># that fold-change values of 2 are the min value</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">DEGs.alpha</span><span class="op">[</span> , <span class="fl">3</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>,<span class="fl">2</span>,<span class="va">range</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           S1&lt;-&gt;S2     S1&lt;-&gt;S3     S2&lt;-&gt;S3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1,] 0.0004200748 0.002629415 0.005515936</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [2,] 0.2405307541 0.046734476 0.375736681</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># now check whether each example has at least one stage with a p-value &lt;= 0.05</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">DEGs.alpha</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    Phylostratum      GeneID     S1&lt;-&gt;S2     S1&lt;-&gt;S3    S2&lt;-&gt;S3</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3             1 at1g01070.1 0.007037106 0.002629415 0.01981887</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5             1 at1g01090.1 0.026369854 0.035285481 0.08957681</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 31            1 at1g01770.1 0.240530754 0.012040556 0.04951844</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 45            1 at1g02130.1 0.008556501 0.046734476 0.32347638</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 47            1 at1g02205.3 0.022021429 0.015537681 0.22278162</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 93            1 at1g03190.1 0.016635354 0.012251604 0.14721584</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Nikola Kalábová, Kristian K Ullrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

