<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Differential Gene Expression Analysis — DiffGenes • myTAI</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Differential Gene Expression Analysis — DiffGenes" />
<meta property="og:description" content="Detect differentially expressed genes (DEGs) in a standard ExpressionSet object." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI package</a>
    </li>
    <li>
      <a href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/drostlab/myTAI/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Differential Gene Expression Analysis</h1>
    <small class="dont-index">Source: <a href='https://github.com/drostlab/myTAI/blob/master/R/DiffGenes.R'><code>R/DiffGenes.R</code></a></small>
    <div class="hidden name"><code>DiffGenes.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Detect differentially expressed genes (DEGs) in a standard <code>ExpressionSet</code> object.</p>
    </div>

    <pre class="usage"><span class='fu'>DiffGenes</span><span class='op'>(</span>
  <span class='va'>ExpressionSet</span>,
  <span class='va'>nrep</span>,
  method <span class='op'>=</span> <span class='st'>"foldchange"</span>,
  lib.size <span class='op'>=</span> <span class='cn'>NULL</span>,
  p.adjust.method <span class='op'>=</span> <span class='cn'>NULL</span>,
  comparison <span class='op'>=</span> <span class='cn'>NULL</span>,
  alpha <span class='op'>=</span> <span class='cn'>NULL</span>,
  filter.method <span class='op'>=</span> <span class='cn'>NULL</span>,
  n <span class='op'>=</span> <span class='cn'>NULL</span>,
  stage.names <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExpressionSet</th>
      <td><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></td>
    </tr>
    <tr>
      <th>nrep</th>
      <td><p>either a numeric value specifying the constant number of replicates per stage or a numeric vector specifying the variable number of replicates for each stage position.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>method to detect differentially expressed genes.</p></td>
    </tr>
    <tr>
      <th>p.adjust.method</th>
      <td><p>p value correction method that is passed to <code><a href='https://rdrr.io/r/stats/p.adjust.html'>p.adjust</a></code>.
Available options are:</p><ul>
<li><p><code>p.adjust.method = "BH"</code> (Benjamini-Hochberg correction)</p></li>
<li><p><code>p.adjust.method = "bonferroni"</code> (Bonferroni correction)</p></li>
<li><p><code>p.adjust.method = "holm"</code></p></li>
<li><p><code>p.adjust.method = "hochberg"</code></p></li>
<li><p><code>p.adjust.method = "hommel"</code></p></li>
<li><p><code>p.adjust.method = "BY"</code></p></li>
<li><p><code>p.adjust.method = "fdr"</code></p></li>
</ul><p>If <code>p.adjust.method = NULL</code> (Default) then no p-value correction is performed.</p></td>
    </tr>
    <tr>
      <th>comparison</th>
      <td><p>a character string specifying whether genes having fold-change or p-values
 below, above, or below AND above (both) the <code>alpha</code> value should be excluded from the dataset.
 In case <code>comparison = "both"</code> is chosen, the <code>cut.off</code> argument must be a two dimensional vector defining the lower <code>alpha</code> value at the first position and the upper <code>alpha</code> value
at the second position.</p></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p>a numeric value specifying the cut-off value above which Genes fulfilling the corresponding fold-change, log-fold-change, or p-value should be retained and returned by <code>DiffGenes</code>.</p></td>
    </tr>
    <tr>
      <th>filter.method</th>
      <td><p>a method how to <code>alpha</code> values in multiple stages. Options are <code>"const"</code>, <code>"min-set"</code>, and <code>"n-set"</code>.</p></td>
    </tr>
    <tr>
      <th>n</th>
      <td><p>a numeric value for <code>method = "n-set"</code>.</p></td>
    </tr>
    <tr>
      <th>stage.names</th>
      <td><p>a character vector specifying the new names of collapsed stages.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>All methods to perform detection of differentially expressed genes assume that your input
dataset has been normalized before passing it to <em>DiffGenes</em>. For RNA-Seq data
<em>DiffGenes</em> assumes that the libraries have been normalized to have the same size, i.e.,
to have the same expected column sum under the null hypothesis.</p>
<p>Available methods for the detection of differentially expressed genes:</p>
<ul>
<li><p><code>method = "foldchange"</code>: ratio of replicate geometric means between developmental stages.
 Here, the <em>DiffGenes</em> functions assumes that absolute expression levels are stored in your input <code>ExpresisonSet</code>.</p></li>
<li><p><code>method = "log-foldchange"</code>: difference of replicate arithmetic means between developmental stages. Here, the <em>DiffGenes</em> functions assumes that <em>log2a</em>
transformed expression levels are stored in your input <code>ExpresisonSet</code>.</p></li>
<li><p><code>method = "t.test"</code>: Welch t.test between replicate expression levels of two samples.</p></li>
<li><p><code>method = "wilcox.test"</code>: Wilcoxon Rank Sum Test between replicate expression levels of two samples.</p></li>
</ul>

<p>Exclude non differentially expressed genes from the result dataset:</p>
<p>When specifying the <code>alpha</code> argument you furthermore, need to specify the <code>filter.method</code> to decide how non differentially expressed genes should be classified in multiple sample comparisons and which genes should be retained in the final dataset returned by <code>DiffGenes</code>. In other words, all genes &lt; <code>alpha</code> based on the following <code>filter.method</code> are removed from the result dataset.</p>
<p>Following extraction criteria are implemented in this function:</p>
<ul>
<li><p><code>const</code>: all genes that have at least one sample comparison that undercuts or exceeds the <code>alpha</code> value <code>cut.off</code> will be excluded from the <code>ExpressionSet</code>. Hence, for a 7 stage <code>ExpressionSet</code> genes passing the <code>alpha</code> threshold in 6 stages will be retained in the <code>ExpressionSet</code>.</p></li>
<li><p><code>min-set</code>: genes passing the <code>alpha</code> value in <code><a href='https://rdrr.io/r/base/Round.html'>ceiling(n/2)</a></code> stages will be retained in the <code>ExpressionSet</code>, where <em>n</em> is the number of stages in the <code>ExpressionSet</code>.</p></li>
<li><p><code>n-set</code>: genes passing the <code>alpha</code> value in <code>n</code> stages will be retained in the <code>ExpressionSet</code>. Here, the argument <code>n</code> needs to be specified.</p></li>
</ul>

    <h2 class="hasAnchor" id="note"><a class="anchor" href="#note"></a>Note</h2>

    <p>In case input <code>ExpressionSet</code> objects store 0 values, internally all expression levels are 
shifted by <code>+1</code> to allow sufficient fold-change and p-value computations. Additionally, a warning
is printed to the console in case expression levels have been automatically shifted.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='Expressed.html'>Expressed</a></code></p></div>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Hajk-Georg Drost</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>PhyloExpressionSetExample</span><span class='op'>)</span>

<span class='co'># Detection of DEGs using the fold-change measure</span>
<span class='va'>DEGs</span> <span class='op'>&lt;-</span> <span class='fu'>DiffGenes</span><span class='op'>(</span>ExpressionSet <span class='op'>=</span> <span class='va'>PhyloExpressionSetExample</span><span class='op'>[</span> ,<span class='fl'>1</span><span class='op'>:</span><span class='fl'>8</span><span class='op'>]</span>,
                  nrep          <span class='op'>=</span> <span class='fl'>2</span>,
                  comparison    <span class='op'>=</span> <span class='st'>"below"</span>,
                  method        <span class='op'>=</span> <span class='st'>"foldchange"</span>,
                  stage.names   <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"S1"</span>,<span class='st'>"S2"</span>,<span class='st'>"S3"</span><span class='op'>)</span><span class='op'>)</span>


<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>DEGs</span><span class='op'>)</span>
</div><div class='output co'>#&gt;   Phylostratum      GeneID    S1-&gt;S2    S1-&gt;S3    S2-&gt;S1    S2-&gt;S3    S3-&gt;S1
#&gt; 1            1 at1g01040.2 1.6706342 2.0767465 0.5985751 1.2430887 0.4815224
#&gt; 2            1 at1g01050.1 1.0231405 1.2788699 0.9773829 1.2499456 0.7819403
#&gt; 3            1 at1g01070.1 1.3087104 1.4044710 0.7641110 1.0731717 0.7120119
#&gt; 4            1 at1g01080.2 0.7786521 0.7286129 1.2842706 0.9357362 1.3724709
#&gt; 5            1 at1g01090.1 0.3744682 0.2255944 2.6704536 0.6024393 4.4327345
#&gt; 6            1 at1g01120.1 0.9110850 0.9000931 1.0975925 0.9879354 1.1109962
#&gt;      S3-&gt;S2
#&gt; 1 0.8044478
#&gt; 2 0.8000348
#&gt; 3 0.9318173
#&gt; 4 1.0686773
#&gt; 5 1.6599182
#&gt; 6 1.0122119</div><div class='input'>

<span class='co'># Detection of DEGs using the log-fold-change measure</span>
<span class='co'># when choosing method = "log-foldchange" it is assumed that</span>
<span class='co'># your input expression matrix stores log2 expression levels </span>
<span class='va'>log.DEGs</span> <span class='op'>&lt;-</span> <span class='fu'>DiffGenes</span><span class='op'>(</span>ExpressionSet <span class='op'>=</span> <span class='fu'><a href='tf.html'>tf</a></span><span class='op'>(</span><span class='va'>PhyloExpressionSetExample</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>5</span>,<span class='fl'>1</span><span class='op'>:</span><span class='fl'>8</span><span class='op'>]</span>,<span class='va'>log2</span><span class='op'>)</span>,
                      nrep          <span class='op'>=</span> <span class='fl'>2</span>,
                      comparison    <span class='op'>=</span> <span class='st'>"below"</span>,
                      method        <span class='op'>=</span> <span class='st'>"log-foldchange"</span>,
                      stage.names   <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"S1"</span>,<span class='st'>"S2"</span>,<span class='st'>"S3"</span><span class='op'>)</span><span class='op'>)</span>


<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>log.DEGs</span><span class='op'>)</span>
</div><div class='output co'>#&gt;   Phylostratum      GeneID      S1-&gt;S2     S1-&gt;S3      S2-&gt;S1      S2-&gt;S3
#&gt; 1            1 at1g01040.2  0.74039587  1.0543251 -0.74039587  0.31392926
#&gt; 2            1 at1g01050.1  0.03300429  0.3548696 -0.03300429  0.32186526
#&gt; 3            1 at1g01070.1  0.38814592  0.4900268 -0.38814592  0.10188092
#&gt; 4            1 at1g01080.2 -0.36094927 -0.4567755  0.36094927 -0.09582626
#&gt; 5            1 at1g01090.1 -1.41708484 -2.1481970  1.41708484 -0.73111211
#&gt;       S3-&gt;S1      S3-&gt;S2
#&gt; 1 -1.0543251 -0.31392926
#&gt; 2 -0.3548696 -0.32186526
#&gt; 3 -0.4900268 -0.10188092
#&gt; 4  0.4567755  0.09582626
#&gt; 5  2.1481970  0.73111211</div><div class='input'>

<span class='co'># Remove fold-change values &lt; 2 from the dataset:</span>

<span class='co'>## first have a look at the range of fold-change values of all genes </span>
<span class='fu'><a href='https://rdrr.io/r/base/apply.html'>apply</a></span><span class='op'>(</span><span class='va'>DEGs</span><span class='op'>[</span> , <span class='fl'>3</span><span class='op'>:</span><span class='fl'>8</span><span class='op'>]</span>,<span class='fl'>2</span>,<span class='va'>range</span><span class='op'>)</span>
</div><div class='output co'>#&gt;          S1-&gt;S2      S1-&gt;S3     S2-&gt;S1      S2-&gt;S3     S3-&gt;S1      S3-&gt;S2
#&gt; [1,]  0.1081031  0.04752967 0.06022388  0.06115996  0.0244683  0.06450549
#&gt; [2,] 16.6047090 40.86920354 9.25042630 15.50255551 21.0394884 16.35056753</div><div class='input'>
<span class='co'># now remove genes undercutting the alpha = 2 threshold</span>
<span class='co'># hence, remove genes having p-values &lt;= 0.05 in at</span>
<span class='co'># least one sample comparison</span>
<span class='va'>DEGs.alpha</span> <span class='op'>&lt;-</span> <span class='fu'>DiffGenes</span><span class='op'>(</span>ExpressionSet <span class='op'>=</span> <span class='va'>PhyloExpressionSetExample</span><span class='op'>[</span><span class='fl'>1</span><span class='op'>:</span><span class='fl'>250</span> ,<span class='fl'>1</span><span class='op'>:</span><span class='fl'>8</span><span class='op'>]</span>,
                        nrep          <span class='op'>=</span> <span class='fl'>2</span>,
                        method        <span class='op'>=</span> <span class='st'>"t.test"</span>,
                        alpha         <span class='op'>=</span> <span class='fl'>0.05</span>,
                        comparison    <span class='op'>=</span> <span class='st'>"above"</span>,
                        filter.method <span class='op'>=</span> <span class='st'>"n-set"</span>,
                        n             <span class='op'>=</span> <span class='fl'>1</span>,
                        stage.names   <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='st'>"S1"</span>,<span class='st'>"S2"</span>,<span class='st'>"S3"</span><span class='op'>)</span><span class='op'>)</span>

<span class='co'># now again have a look at the range and find</span>
<span class='co'># that fold-change values of 2 are the min value</span>
<span class='fu'><a href='https://rdrr.io/r/base/apply.html'>apply</a></span><span class='op'>(</span><span class='va'>DEGs.alpha</span><span class='op'>[</span> , <span class='fl'>3</span><span class='op'>:</span><span class='fl'>5</span><span class='op'>]</span>,<span class='fl'>2</span>,<span class='va'>range</span><span class='op'>)</span>
</div><div class='output co'>#&gt;           S1&lt;-&gt;S2     S1&lt;-&gt;S3     S2&lt;-&gt;S3
#&gt; [1,] 0.0004200748 0.002629415 0.005515936
#&gt; [2,] 0.2405307541 0.046734476 0.375736681</div><div class='input'>
<span class='co'># now check whether each example has at least one stage with a p-value &lt;= 0.05</span>
<span class='fu'><a href='https://rdrr.io/r/utils/head.html'>head</a></span><span class='op'>(</span><span class='va'>DEGs.alpha</span><span class='op'>)</span>
</div><div class='output co'>#&gt;    Phylostratum      GeneID     S1&lt;-&gt;S2     S1&lt;-&gt;S3    S2&lt;-&gt;S3
#&gt; 3             1 at1g01070.1 0.007037106 0.002629415 0.01981887
#&gt; 5             1 at1g01090.1 0.026369854 0.035285481 0.08957681
#&gt; 31            1 at1g01770.1 0.240530754 0.012040556 0.04951844
#&gt; 45            1 at1g02130.1 0.008556501 0.046734476 0.32347638
#&gt; 47            1 at1g02205.3 0.022021429 0.015537681 0.22278162
#&gt; 93            1 at1g03190.1 0.016635354 0.012251604 0.14721584</div><div class='input'>
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


