<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Perform Reductive Early Conservation Test — EarlyConservationTest • myTAI</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Perform Reductive Early Conservation Test — EarlyConservationTest" />
<meta property="og:description" content="The Reductive Early Conservation Test aims to statistically evaluate the
existence of a monotonically increasing phylotranscriptomic pattern based on TAI or TDI computations.
The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern) 
does not follow an early conservation like pattern. A p-value &amp;lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does
indeed follow an early conservation (low-high-high) shape." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI package</a>
    </li>
    <li>
      <a href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/drostlab/myTAI/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform Reductive Early Conservation Test</h1>
    <small class="dont-index">Source: <a href='https://github.com/drostlab/myTAI/blob/master/R/EarlyConservationTest.R'><code>R/EarlyConservationTest.R</code></a></small>
    <div class="hidden name"><code>EarlyConservationTest.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The <em>Reductive Early Conservation Test</em> aims to statistically evaluate the
existence of a monotonically increasing phylotranscriptomic pattern based on <code><a href='TAI.html'>TAI</a></code> or <code><a href='TDI.html'>TDI</a></code> computations.
The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern) 
does not follow an early conservation like pattern. A p-value &lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does
indeed follow an early conservation (low-high-high) shape.</p>
    </div>

    <pre class="usage"><span class='fu'>EarlyConservationTest</span><span class='op'>(</span>
  <span class='va'>ExpressionSet</span>,
  modules <span class='op'>=</span> <span class='cn'>NULL</span>,
  permutations <span class='op'>=</span> <span class='fl'>1000</span>,
  lillie.test <span class='op'>=</span> <span class='cn'>FALSE</span>,
  plotHistogram <span class='op'>=</span> <span class='cn'>FALSE</span>,
  runs <span class='op'>=</span> <span class='fl'>10</span>,
  parallel <span class='op'>=</span> <span class='cn'>FALSE</span>,
  gof.warning <span class='op'>=</span> <span class='cn'>FALSE</span>,
  custom.perm.matrix <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExpressionSet</th>
      <td><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></td>
    </tr>
    <tr>
      <th>modules</th>
      <td><p>a list storing three elements: early, mid, and late. Each element expects a numeric
vector specifying the developmental stages or experiments that correspond to each module. 
For example, <code>module</code> = list(early = 1:2, mid = 3:5, late = 6:7) devides a dataset 
storing seven developmental stages into 3 modules.</p></td>
    </tr>
    <tr>
      <th>permutations</th>
      <td><p>a numeric value specifying the number of permutations to be performed for the <code>ReductiveHourglassTest</code>.</p></td>
    </tr>
    <tr>
      <th>lillie.test</th>
      <td><p>a boolean value specifying whether the Lilliefors Kolmogorov-Smirnov Test shall be performed to quantify the goodness of fit.</p></td>
    </tr>
    <tr>
      <th>plotHistogram</th>
      <td><p>a boolean value specifying whether a <em>Lillifor's Kolmogorov-Smirnov-Test</em> 
shall be performed to test the goodness of fit of the approximated distribution, as well as additional plots quantifying the significance
of the observed phylotranscriptomic pattern.</p></td>
    </tr>
    <tr>
      <th>runs</th>
      <td><p>specify the number of runs to be performed for goodness of fit computations, in case <code>plotHistogram</code> = <code>TRUE</code>.
In most cases <code>runs</code> = 100 is a reasonable choice. Default is <code>runs</code> = 10 (because it takes less computation time for demonstration purposes).</p></td>
    </tr>
    <tr>
      <th>parallel</th>
      <td><p>performing <code>runs</code> in parallel (takes all cores of your multicore machine).</p></td>
    </tr>
    <tr>
      <th>gof.warning</th>
      <td><p>a logical value indicating whether non significant goodness of fit results should be printed as warning. Default is <code>gof.warning = FALSE</code>.</p></td>
    </tr>
    <tr>
      <th>custom.perm.matrix</th>
      <td><p>a custom <code><a href='bootMatrix.html'>bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic. Default is <code>custom.perm.matrix = NULL</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>a list object containing the list elements:</p>
<p><code>p.value</code> : the p-value quantifying the statistical significance (low-high-high pattern) of the given phylotranscriptomics pattern.</p>
<p><code>std.dev</code> : the standard deviation of the N sampled phylotranscriptomics patterns for each developmental stage S.</p>
<p><code>lillie.test</code> : a boolean value specifying whether the <em>Lillifors KS-Test</em> returned a p-value &gt; 0.05, 
which indicates that fitting the permuted scores with a normal distribution seems plausible.</p>
    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>The <em>reductive early conservation test</em> is a permutation test based on the following test statistic.</p>
<p>(1) A set of developmental stages is partitioned into three modules - early, mid, and late - based on prior biological knowledge.</p>
<p>(2) The mean <code><a href='TAI.html'>TAI</a></code> or <code><a href='TDI.html'>TDI</a></code> value for each of the three modules T_early, T_mid, and T_late are computed.</p>
<p>(3) The two differences D1 = T_mid - T_early and D2 = T_late - T_early are calculated.</p>
<p>(4) The minimum D_min of D1 and D2 is computed as final test statistic of the reductive hourglass test.</p>

<p>In order to determine the statistical significance of an observed minimum difference D_min 
the following permutation test was performed. Based on the <code><a href='bootMatrix.html'>bootMatrix</a></code> D_min 
is calculated from each of the permuted <code><a href='TAI.html'>TAI</a></code> or <code><a href='TDI.html'>TDI</a></code> profiles, 
approximated by a Gaussian distribution with method of moments estimated parameters returned by <code><a href='https://rdrr.io/pkg/fitdistrplus/man/fitdist.html'>fitdist</a></code>, 
and the corresponding p-value is computed by <code><a href='https://rdrr.io/r/stats/Normal.html'>pnorm</a></code> given the estimated parameters of the Gaussian distribution. 
The <em>goodness of fit</em> for the random vector <em>D_min</em> is statistically quantified by an Lilliefors (Kolmogorov-Smirnov) test 
for normality.</p>

<p>In case the parameter <em>plotHistogram = TRUE</em>, a multi-plot is generated showing:</p>       
<p>(1) A Cullen and Frey skewness-kurtosis plot generated by <code><a href='https://rdrr.io/pkg/fitdistrplus/man/descdist.html'>descdist</a></code>. 
This plot illustrates which distributions seem plausible to fit the resulting permutation vector D_min. 
In the case of the <em>reductive early conservation test</em> a normal distribution seemed plausible.</p>
<p>(2) A histogram of D_min combined with the density plot is plotted. D_min is then fitted by a normal distribution. 
The corresponding parameters are estimated by <em>moment matching estimation</em> using the <code><a href='https://rdrr.io/pkg/fitdistrplus/man/fitdist.html'>fitdist</a></code> function.</p>
<p>(3) A plot showing the p-values for N independent runs to verify that a specific p-value is biased by a specific permutation order.</p>
<p>(4) A barplot showing the number of cases in which the underlying goodness of fit (returned by Lilliefors (Kolmogorov-Smirnov) test 
for normality) has shown to be significant (<code>TRUE</code>) or not significant (<code>FALSE</code>). 
This allows to quantify the permutation bias and their implications on the goodness of fit.</p>
    <h2 class="hasAnchor" id="references"><a class="anchor" href="#references"></a>References</h2>

    <p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012</p>
<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.</p>
<p>Piasecka B, Lichocki P, Moretti S, et al. (2013) <em>The hourglass and the early conservation models co-existing
patterns of developmental constraints in vertebrates</em>. PLoS Genet. 9(4): e1003476.</p>
    <h2 class="hasAnchor" id="see-also"><a class="anchor" href="#see-also"></a>See also</h2>

    <div class='dont-index'><p><code><a href='ecScore.html'>ecScore</a></code>, <code><a href='bootMatrix.html'>bootMatrix</a></code>, <code><a href='FlatLineTest.html'>FlatLineTest</a></code>,<code><a href='ReductiveHourglassTest.html'>ReductiveHourglassTest</a></code>, <code><a href='ReverseHourglassTest.html'>ReverseHourglassTest</a></code>, <code><a href='PlotSignature.html'>PlotSignature</a></code></p></div>
    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Hajk-Georg Drost</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'>
<span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>PhyloExpressionSetExample</span><span class='op'>)</span>

<span class='co'># perform the early conservation test for a PhyloExpressionSet</span>
<span class='co'># here the prior biological knowledge is that stages 1-2 correspond to module 1 = early,</span>
<span class='co'># stages 3-5 to module 2 = mid (phylotypic module), and stages 6-7 correspond to</span>
<span class='co'># module 3 = late</span>
<span class='fu'>EarlyConservationTest</span><span class='op'>(</span><span class='va'>PhyloExpressionSetExample</span>,
                       modules <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>early <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span>, mid <span class='op'>=</span> <span class='fl'>3</span><span class='op'>:</span><span class='fl'>5</span>, late <span class='op'>=</span> <span class='fl'>6</span><span class='op'>:</span><span class='fl'>7</span><span class='op'>)</span>, 
                       permutations <span class='op'>=</span> <span class='fl'>1000</span><span class='op'>)</span>
</div><div class='output co'>#&gt; $start.arg
#&gt; $start.arg$mean
#&gt; [1] -0.01365784
#&gt; 
#&gt; $start.arg$sd
#&gt; [1] 0.03050271
#&gt; 
#&gt; 
#&gt; $fix.arg
#&gt; NULL
#&gt; </div><div class='output co'>#&gt; $p.value
#&gt; [1] 0.9999168
#&gt; 
#&gt; $std.dev
#&gt; [1] 0.05547201 0.05425241 0.05199121 0.05117642 0.05070018 0.05318647 0.05791163
#&gt; 
#&gt; $lillie.test
#&gt; [1] NA
#&gt; </div><div class='input'>

<span class='co'># use your own permutation matrix based on which p-values (EarlyConservationTest)</span>
<span class='co'># shall be computed</span>
<span class='va'>custom_perm_matrix</span> <span class='op'>&lt;-</span> <span class='fu'><a href='bootMatrix.html'>bootMatrix</a></span><span class='op'>(</span><span class='va'>PhyloExpressionSetExample</span>,<span class='fl'>100</span><span class='op'>)</span>

<span class='fu'>EarlyConservationTest</span><span class='op'>(</span><span class='va'>PhyloExpressionSetExample</span>,
                       modules <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/list.html'>list</a></span><span class='op'>(</span>early <span class='op'>=</span> <span class='fl'>1</span><span class='op'>:</span><span class='fl'>2</span>, mid <span class='op'>=</span> <span class='fl'>3</span><span class='op'>:</span><span class='fl'>5</span>, late <span class='op'>=</span> <span class='fl'>6</span><span class='op'>:</span><span class='fl'>7</span><span class='op'>)</span>, 
                       custom.perm.matrix <span class='op'>=</span> <span class='va'>custom_perm_matrix</span><span class='op'>)</span>
</div><div class='output co'>#&gt; $start.arg
#&gt; $start.arg$mean
#&gt; [1] -0.01701384
#&gt; 
#&gt; $start.arg$sd
#&gt; [1] 0.03440646
#&gt; 
#&gt; 
#&gt; $fix.arg
#&gt; NULL
#&gt; </div><div class='output co'>#&gt; $p.value
#&gt; [1] 0.9994032
#&gt; 
#&gt; $std.dev
#&gt;     Zygote   Quadrant   Globular      Heart    Torpedo       Bent     Mature 
#&gt; 0.05953318 0.05925106 0.05587577 0.05336733 0.05550997 0.05758198 0.06123089 
#&gt; 
#&gt; $lillie.test
#&gt; [1] NA
#&gt; </div><div class='input'>                       
</div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


