<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Filter for Expressed Genes — Expressed • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Filter for Expressed Genes — Expressed"><meta name="description" content="This function takes an ExpressionSet object and removes genes from the gene expression matrix that
have an expression level below, above, or below AND above a defined cut.off value. Hence, this function allows to remove
genes that have been defined as not expressed or outliers and returns an ExpressionSet retaining only expressed genes."><meta property="og:description" content="This function takes an ExpressionSet object and removes genes from the gene expression matrix that
have an expression level below, above, or below AND above a defined cut.off value. Hence, this function allows to remove
genes that have been defined as not expressed or outliers and returns an ExpressionSet retaining only expressed genes."><meta property="og:image" content="https://drostlab.github.io/myTAI/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Enrichment.html">Enrichment Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a></li>
    <li><a class="dropdown-item" href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Introduction.html">Introduction to the myTAI package</a></li>
    <li><a class="dropdown-item" href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Filter for Expressed Genes</h1>
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/Expressed.R" class="external-link"><code>R/Expressed.R</code></a></small>
      <div class="d-none name"><code>Expressed.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function takes an ExpressionSet object and removes genes from the gene expression matrix that
have an expression level below, above, or below AND above a defined <code>cut.off</code> value. Hence, this function allows to remove
genes that have been defined as <em>not expressed</em> or <em>outliers</em> and returns an <code>ExpressionSet</code> retaining only expressed genes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">Expressed</span><span class="op">(</span></span>
<span>  <span class="va">ExpressionSet</span>,</span>
<span>  <span class="va">cut.off</span>,</span>
<span>  method <span class="op">=</span> <span class="st">"const"</span>,</span>
<span>  comparison <span class="op">=</span> <span class="st">"below"</span>,</span>
<span>  n <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-expressionset">ExpressionSet<a class="anchor" aria-label="anchor" href="#arg-expressionset"></a></dt>
<dd><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></dd>


<dt id="arg-cut-off">cut.off<a class="anchor" aria-label="anchor" href="#arg-cut-off"></a></dt>
<dd><p>a numeric value specifying the expression cut-off to define genes as <em>not expressed</em> (<code>comparison = "below"</code>) , <em>outliers</em> (<code>comparison = "above"</code>), or both (<code>comparison = "both"</code>). See <code>comparison</code> for details. In case <code>comparison = "both"</code>, the <code>cut.off</code> argument must be a two dimensional vector defining the lower <code>cut.off</code> value at the first position and the upper <code>cut.off</code> value at the second position.</p></dd>


<dt id="arg-method">method<a class="anchor" aria-label="anchor" href="#arg-method"></a></dt>
<dd><p>a method defining how to treat gene expression values in multiple stages. The corresponding method that is chosen allows to control the stage-wise fulfillment of the threshold criteria. Options are <code>"const"</code>, <code>"min-set"</code>, and <code>"n-set"</code>.</p></dd>


<dt id="arg-comparison">comparison<a class="anchor" aria-label="anchor" href="#arg-comparison"></a></dt>
<dd><p>a character string specifying whether genes having expression levels
 below, above, or below AND above (both) the <code>cut.off</code> value should be excluded from the dataset.
 In case <code>comparison = "both"</code> is chosen, the <code>cut.off</code> argument must be a two dimensional vector defining the lower <code>cut.off</code> value at the first position and the upper <code>cut.off</code> value
at the second position.</p></dd>


<dt id="arg-n">n<a class="anchor" aria-label="anchor" href="#arg-n"></a></dt>
<dd><p>a numeric value for <code>method = "n-set"</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This filter function allows users to remove genes from the <code>ExpressionSet</code> object that undercut or exceed a certain expression level <code>cut.off</code>.</p>
<p>Following extraction criteria are implemented in this function:</p><ul><li><p><code>const</code>: all genes that have at least one stage that undercuts or exceeds the expression <code>cut.off</code> will be excluded from the <code>ExpressionSet</code>. Hence, for a 7 stage <code>ExpressionSet</code> genes passing the expression level <code>cut.off</code> in 6 stages will be retained in the <code>ExpressionSet</code>.</p></li>
<li><p><code>min-set</code>: genes passing the expression level <code>cut.off</code> in <code>ceiling(n/2)</code> stages will be retained in the <code>ExpressionSet</code>, where <em>n</em> is the number of stages in the <code>ExpressionSet</code>.</p></li>
<li><p><code>n-set</code>: genes passing the expression level <code>cut.off</code> in <code>n</code> stages will be retained in the <code>ExpressionSet</code>. Here, the argument <code>n</code> needs to be specified.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># remove genes that have an expression level below 8000 </span></span></span>
<span class="r-in"><span><span class="co"># in at least one developmental stage</span></span></span>
<span class="r-in"><span><span class="va">FilterConst</span> <span class="op">&lt;-</span> <span class="fu">Expressed</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>                         cut.off       <span class="op">=</span> <span class="fl">8000</span>, </span></span>
<span class="r-in"><span>                         method        <span class="op">=</span> <span class="st">"const"</span>,</span></span>
<span class="r-in"><span>                         comparison    <span class="op">=</span> <span class="st">"below"</span><span class="op">)</span></span></span>
<span class="r-in"><span>                              </span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">FilterConst</span><span class="op">)</span> <span class="co"># check number of retained genes</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 449   9</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># remove genes that have an expression level below 8000 </span></span></span>
<span class="r-in"><span><span class="co"># in at least 3 developmental stages </span></span></span>
<span class="r-in"><span><span class="co"># (in this case: ceiling(7/2) = 4 stages fulfilling the cut-off criteria)</span></span></span>
<span class="r-in"><span><span class="va">FilterMinSet</span> <span class="op">&lt;-</span> <span class="fu">Expressed</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>                          cut.off       <span class="op">=</span> <span class="fl">8000</span>, </span></span>
<span class="r-in"><span>                          method        <span class="op">=</span> <span class="st">"min-set"</span>,</span></span>
<span class="r-in"><span>                          comparison    <span class="op">=</span> <span class="st">"below"</span><span class="op">)</span></span></span>
<span class="r-in"><span>                               </span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">FilterMinSet</span><span class="op">)</span> <span class="co"># check number of retained genes</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 830   9</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># remove genes that have an expression level below 8000 </span></span></span>
<span class="r-in"><span><span class="co"># in at least 5 developmental stages (in this case: n = 2 stages fulfilling the criteria)</span></span></span>
<span class="r-in"><span><span class="va">FilterNSet</span> <span class="op">&lt;-</span> <span class="fu">Expressed</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>                        cut.off       <span class="op">=</span> <span class="fl">8000</span>, </span></span>
<span class="r-in"><span>                        method        <span class="op">=</span> <span class="st">"n-set"</span>,</span></span>
<span class="r-in"><span>                        comparison    <span class="op">=</span> <span class="st">"below"</span>,</span></span>
<span class="r-in"><span>                        n             <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span>                               </span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">FilterMinSet</span><span class="op">)</span> <span class="co"># check number of retained genes</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 830   9</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># remove expression levels that exceed the cut.off criteria</span></span></span>
<span class="r-in"><span><span class="va">FilterMinSet</span> <span class="op">&lt;-</span> <span class="fu">Expressed</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>                          cut.off       <span class="op">=</span> <span class="fl">12000</span>, </span></span>
<span class="r-in"><span>                          method        <span class="op">=</span> <span class="st">"min-set"</span>,</span></span>
<span class="r-in"><span>                          comparison    <span class="op">=</span> <span class="st">"above"</span><span class="op">)</span></span></span>
<span class="r-in"><span>                               </span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">FilterMinSet</span><span class="op">)</span> <span class="co"># check number of retained genes</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 24456     9</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># remove expression levels that undercut AND exceed the cut.off criteria</span></span></span>
<span class="r-in"><span><span class="va">FilterMinSet</span> <span class="op">&lt;-</span> <span class="fu">Expressed</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>                          cut.off       <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">8000</span>,<span class="fl">12000</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>                          method        <span class="op">=</span> <span class="st">"min-set"</span>,</span></span>
<span class="r-in"><span>                          comparison    <span class="op">=</span> <span class="st">"both"</span><span class="op">)</span></span></span>
<span class="r-in"><span>                               </span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">FilterMinSet</span><span class="op">)</span> <span class="co"># check number of retained genes</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 111   9</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich, Stefan Manolache.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

