<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Filter for Expressed Genes — Expressed • myTAI</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Filter for Expressed Genes — Expressed" />
<meta property="og:description" content="This function takes an ExpressionSet object and removes genes from the gene expression matrix that
have an expression level below, above, or below AND above a defined cut.off value. Hence, this function allows to remove
genes that have been defined as not expressed or outliers and returns an ExpressionSet retaining only expressed genes." />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI package</a>
    </li>
    <li>
      <a href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
    <li>
      <a href="../articles/scRNAseq.html">Sincgle-Cell Expression Analysis with `myTAI`</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/drostlab/myTAI/">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Filter for Expressed Genes</h1>
    <small class="dont-index">Source: <a href='https://github.com/drostlab/myTAI/blob/master/R/Expressed.R'><code>R/Expressed.R</code></a></small>
    <div class="hidden name"><code>Expressed.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function takes an ExpressionSet object and removes genes from the gene expression matrix that
have an expression level below, above, or below AND above a defined <code>cut.off</code> value. Hence, this function allows to remove
genes that have been defined as <em>not expressed</em> or <em>outliers</em> and returns an <code>ExpressionSet</code> retaining only expressed genes.</p>
    </div>

    <pre class="usage"><span class='fu'>Expressed</span><span class='op'>(</span>
  <span class='va'>ExpressionSet</span>,
  <span class='va'>cut.off</span>,
  method <span class='op'>=</span> <span class='st'>"const"</span>,
  comparison <span class='op'>=</span> <span class='st'>"below"</span>,
  n <span class='op'>=</span> <span class='cn'>NULL</span>
<span class='op'>)</span></pre>

    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>ExpressionSet</th>
      <td><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></td>
    </tr>
    <tr>
      <th>cut.off</th>
      <td><p>a numeric value specifying the expression cut-off to define genes as <em>not expressed</em> (<code>comparison = "below"</code>) , <em>outliers</em> (<code>comparison = "above"</code>), or both (<code>comparison = "both"</code>). See <code>comparison</code> for details. In case <code>comparison = "both"</code>, the <code>cut.off</code> argument must be a two dimensional vector defining the lower <code>cut.off</code> value at the first position and the upper <code>cut.off</code> value at the second position.</p></td>
    </tr>
    <tr>
      <th>method</th>
      <td><p>a method defining how to treat gene expression values in multiple stages. The corresponding method that is chosen allows to control the stage-wise fulfillment of the threshold criteria. Options are <code>"const"</code>, <code>"min-set"</code>, and <code>"n-set"</code>.</p></td>
    </tr>
    <tr>
      <th>comparison</th>
      <td><p>a character string specifying whether genes having expression levels
 below, above, or below AND above (both) the <code>cut.off</code> value should be excluded from the dataset.
 In case <code>comparison = "both"</code> is chosen, the <code>cut.off</code> argument must be a two dimensional vector defining the lower <code>cut.off</code> value at the first position and the upper <code>cut.off</code> value
at the second position.</p></td>
    </tr>
    <tr>
      <th>n</th>
      <td><p>a numeric value for <code>method = "n-set"</code>.</p></td>
    </tr>
    </table>

    <h2 class="hasAnchor" id="details"><a class="anchor" href="#details"></a>Details</h2>

    <p>This filter function allows users to remove genes from the <code>ExpressionSet</code> object that undercut or exceed a certain expression level <code>cut.off</code>.</p>
<p>Following extraction criteria are implemented in this function:</p><ul>
<li><p><code>const</code>: all genes that have at least one stage that undercuts or exceeds the expression <code>cut.off</code> will be excluded from the <code>ExpressionSet</code>. Hence, for a 7 stage <code>ExpressionSet</code> genes passing the expression level <code>cut.off</code> in 6 stages will be retained in the <code>ExpressionSet</code>.</p></li>
<li><p><code>min-set</code>: genes passing the expression level <code>cut.off</code> in <code>ceiling(n/2)</code> stages will be retained in the <code>ExpressionSet</code>, where <em>n</em> is the number of stages in the <code>ExpressionSet</code>.</p></li>
<li><p><code>n-set</code>: genes passing the expression level <code>cut.off</code> in <code>n</code> stages will be retained in the <code>ExpressionSet</code>. Here, the argument <code>n</code> needs to be specified.</p></li>
</ul>

    <h2 class="hasAnchor" id="author"><a class="anchor" href="#author"></a>Author</h2>

    <p>Hajk-Georg Drost</p>

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><span class='r-in'><span class='fu'><a href='https://rdrr.io/r/utils/data.html'>data</a></span><span class='op'>(</span><span class='va'>PhyloExpressionSetExample</span><span class='op'>)</span></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># remove genes that have an expression level below 8000 </span></span>
<span class='r-in'><span class='co'># in at least one developmental stage</span></span>
<span class='r-in'><span class='va'>FilterConst</span> <span class='op'>&lt;-</span> <span class='fu'>Expressed</span><span class='op'>(</span>ExpressionSet <span class='op'>=</span> <span class='va'>PhyloExpressionSetExample</span>, </span>
<span class='r-in'>                         cut.off       <span class='op'>=</span> <span class='fl'>8000</span>, </span>
<span class='r-in'>                         method        <span class='op'>=</span> <span class='st'>"const"</span>,</span>
<span class='r-in'>                         comparison    <span class='op'>=</span> <span class='st'>"below"</span><span class='op'>)</span></span>
<span class='r-in'>                              </span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='va'>FilterConst</span><span class='op'>)</span> <span class='co'># check number of retained genes</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] 449   9</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># remove genes that have an expression level below 8000 </span></span>
<span class='r-in'><span class='co'># in at least 3 developmental stages </span></span>
<span class='r-in'><span class='co'># (in this case: ceiling(7/2) = 4 stages fulfilling the cut-off criteria)</span></span>
<span class='r-in'><span class='va'>FilterMinSet</span> <span class='op'>&lt;-</span> <span class='fu'>Expressed</span><span class='op'>(</span>ExpressionSet <span class='op'>=</span> <span class='va'>PhyloExpressionSetExample</span>, </span>
<span class='r-in'>                          cut.off       <span class='op'>=</span> <span class='fl'>8000</span>, </span>
<span class='r-in'>                          method        <span class='op'>=</span> <span class='st'>"min-set"</span>,</span>
<span class='r-in'>                          comparison    <span class='op'>=</span> <span class='st'>"below"</span><span class='op'>)</span></span>
<span class='r-in'>                               </span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='va'>FilterMinSet</span><span class='op'>)</span> <span class='co'># check number of retained genes</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] 830   9</span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># remove genes that have an expression level below 8000 </span></span>
<span class='r-in'><span class='co'># in at least 5 developmental stages (in this case: n = 2 stages fulfilling the criteria)</span></span>
<span class='r-in'><span class='va'>FilterNSet</span> <span class='op'>&lt;-</span> <span class='fu'>Expressed</span><span class='op'>(</span>ExpressionSet <span class='op'>=</span> <span class='va'>PhyloExpressionSetExample</span>, </span>
<span class='r-in'>                        cut.off       <span class='op'>=</span> <span class='fl'>8000</span>, </span>
<span class='r-in'>                        method        <span class='op'>=</span> <span class='st'>"n-set"</span>,</span>
<span class='r-in'>                        comparison    <span class='op'>=</span> <span class='st'>"below"</span>,</span>
<span class='r-in'>                        n             <span class='op'>=</span> <span class='fl'>2</span><span class='op'>)</span></span>
<span class='r-in'>                               </span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='va'>FilterMinSet</span><span class='op'>)</span> <span class='co'># check number of retained genes</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] 830   9</span>
<span class='r-in'></span>
<span class='r-in'></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># remove expression levels that exceed the cut.off criteria</span></span>
<span class='r-in'><span class='va'>FilterMinSet</span> <span class='op'>&lt;-</span> <span class='fu'>Expressed</span><span class='op'>(</span>ExpressionSet <span class='op'>=</span> <span class='va'>PhyloExpressionSetExample</span>, </span>
<span class='r-in'>                          cut.off       <span class='op'>=</span> <span class='fl'>12000</span>, </span>
<span class='r-in'>                          method        <span class='op'>=</span> <span class='st'>"min-set"</span>,</span>
<span class='r-in'>                          comparison    <span class='op'>=</span> <span class='st'>"above"</span><span class='op'>)</span></span>
<span class='r-in'>                               </span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='va'>FilterMinSet</span><span class='op'>)</span> <span class='co'># check number of retained genes</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] 24456     9</span>
<span class='r-in'></span>
<span class='r-in'></span>
<span class='r-in'><span class='co'># remove expression levels that undercut AND exceed the cut.off criteria</span></span>
<span class='r-in'><span class='va'>FilterMinSet</span> <span class='op'>&lt;-</span> <span class='fu'>Expressed</span><span class='op'>(</span>ExpressionSet <span class='op'>=</span> <span class='va'>PhyloExpressionSetExample</span>, </span>
<span class='r-in'>                          cut.off       <span class='op'>=</span> <span class='fu'><a href='https://rdrr.io/r/base/c.html'>c</a></span><span class='op'>(</span><span class='fl'>8000</span>,<span class='fl'>12000</span><span class='op'>)</span>, </span>
<span class='r-in'>                          method        <span class='op'>=</span> <span class='st'>"min-set"</span>,</span>
<span class='r-in'>                          comparison    <span class='op'>=</span> <span class='st'>"both"</span><span class='op'>)</span></span>
<span class='r-in'>                               </span>
<span class='r-in'><span class='fu'><a href='https://rdrr.io/r/base/dim.html'>dim</a></span><span class='op'>(</span><span class='va'>FilterMinSet</span><span class='op'>)</span> <span class='co'># check number of retained genes</span></span>
<span class='r-out co'><span class='r-pr'>#&gt;</span> [1] 111   9</span>
<span class='r-in'></span>
<span class='r-in'></span>
</pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Hajk-Georg Drost, Kristian K Ullrich.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


