<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Perform Flat Line Test — FlatLineTest • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Perform Flat Line Test — FlatLineTest"><meta name="description" content="This function quantifies the statistical significance of an observed phylotranscriptomic pattern. In detail, the Flat Line Test quantifies any significant deviation of an observed phylotranscriptomic pattern from a flat line."><meta property="og:description" content="This function quantifies the statistical significance of an observed phylotranscriptomic pattern. In detail, the Flat Line Test quantifies any significant deviation of an observed phylotranscriptomic pattern from a flat line."><meta property="og:image" content="https://drostlab.github.io/myTAI/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Enrichment.html">Enrichment Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a></li>
    <li><a class="dropdown-item" href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Introduction.html">Introduction to the myTAI package</a></li>
    <li><a class="dropdown-item" href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Perform Flat Line Test</h1>
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/FlatLineTest.R" class="external-link"><code>R/FlatLineTest.R</code></a></small>
      <div class="d-none name"><code>FlatLineTest.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function quantifies the statistical significance of an observed phylotranscriptomic pattern. In detail, the <em>Flat Line Test</em> quantifies any significant deviation of an observed phylotranscriptomic pattern from a flat line.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">FlatLineTest</span><span class="op">(</span></span>
<span>  <span class="va">ExpressionSet</span>,</span>
<span>  permutations <span class="op">=</span> <span class="fl">10000</span>,</span>
<span>  plotHistogram <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  runs <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  parallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  custom.perm.matrix <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-expressionset">ExpressionSet<a class="anchor" aria-label="anchor" href="#arg-expressionset"></a></dt>
<dd><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></dd>


<dt id="arg-permutations">permutations<a class="anchor" aria-label="anchor" href="#arg-permutations"></a></dt>
<dd><p>a numeric value specifying the number of permutations that shall be performed for the <em>FlatLineTest</em>.</p></dd>


<dt id="arg-plothistogram">plotHistogram<a class="anchor" aria-label="anchor" href="#arg-plothistogram"></a></dt>
<dd><p>a logical value indicating whether a detailed statistical analysis concerning the goodness of fit should be performed.</p></dd>


<dt id="arg-runs">runs<a class="anchor" aria-label="anchor" href="#arg-runs"></a></dt>
<dd><p>specify the number of runs to be performed for goodness of fit computations. In most cases runs = 100 is a reasonable choice.</p></dd>


<dt id="arg-parallel">parallel<a class="anchor" aria-label="anchor" href="#arg-parallel"></a></dt>
<dd><p>performing <code>runs</code> in parallel (takes all cores of your multicore machine).</p></dd>


<dt id="arg-custom-perm-matrix">custom.perm.matrix<a class="anchor" aria-label="anchor" href="#arg-custom-perm-matrix"></a></dt>
<dd><p>a custom <code><a href="bootMatrix.html">bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic. Default is <code>custom.perm.matrix = NULL</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>a list object containing the list elements:</p><ul><li><p><code>p.value</code> the p-value quantifying the statistical significance (deviation from a flat line) of the given phylotranscriptomics pattern.</p></li>
<li><p><code>std.dev</code> the standard deviation of the N sampled phylotranscriptomics patterns for each developmental stage S.</p></li>
<li><p><code>std.dev</code> the Kolmogorov-Smirnov test satistics for fitting a gamma distribution to the variances of the dataset with permuted phylostrata.</p></li>
</ul></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Internally the function performs N phylotranscriptomics pattern computations (<code><a href="TAI.html">TAI</a></code> or <code><a href="TDI.html">TDI</a></code>) based on sampled PhyloExpressionSets or DivergenceExpressionSets (see <code><a href="bootMatrix.html">bootMatrix</a></code>).
The test statistics is being developed as follows:</p>
<p>The variance <em>V_pattern</em> of the S phylotranscriptomics values defines the test statistic for the <code>FlatLineTest</code>.
The basic assumption is, that the variance of a flat line should be equivalent to zero for a perfect flat line.
Any deviation from a flat line can be measured with a variance value &gt; 0.</p>
<p>To determine the null distribution of <em>V_p</em>, all PS or DS values within each developmental stage s are randomly permuted,
 S surrogate phylotranscriptomics values are computed from this permuted dataset, and a surrogate value of <em>V_p</em> is
  computed from these S phylotranscriptomics values. This permutation process is repeated N times, yielding a histogram of <em>V_p</em>.</p>
<p>After applying a <em>Lilliefors Kolmogorov-Smirnov Test for gamma distribution</em>, <em>V_p</em> is approximated by a <em>gamma distribution</em>.
The two parameters of the <em>gamma distribution</em> are estimated by the function <code><a href="https://lbbe-software.github.io/fitdistrplus/reference/fitdist.html" class="external-link">fitdist</a></code> from the <span class="pkg">fitdistrplus</span> package by <em>moment matching estimation</em>.
The fitted <em>gamma distribution</em> is considered the null distribution of <em>V_pattern</em>, and the p-value of the observed value of <em>V_p</em>
is computed from this null distribution.</p>
<p>In case the parameter <em>plotHistogram = TRUE</em>, a multi-plot is generated showing:</p>
<p>(1) A Cullen and Frey skewness-kurtosis plot generated by <code><a href="https://lbbe-software.github.io/fitdistrplus/reference/descdist.html" class="external-link">descdist</a></code>).</p>
<p>(2) A histogram of V_p combined with the density plot using the Method of Moments estimated parameters returned by the <code><a href="https://lbbe-software.github.io/fitdistrplus/reference/fitdist.html" class="external-link">fitdist</a></code> function using a gamma distribution.</p>
<p>(3) A plot showing the p-values for N independent runs to verify that a specific p-value is biased by a specific permutation order.</p>
<p>The <em>goodness of fit</em> for the random vector <em>V_p</em> is quantified statistically by an adapted Lilliefors (Kolmogorov-Smirnov) test for gamma distributions.</p>
    </div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>In case there are extreme outlier expression values stored in the dataset (PhyloExpressionSet or DivergenceExpressionSet),
 the internal <code><a href="https://lbbe-software.github.io/fitdistrplus/reference/fitdist.html" class="external-link">fitdist</a></code> function that is based on the <code><a href="bootMatrix.html">bootMatrix</a></code> output might return a warning:
  "In densfun(x, parm[1], parm[2], ...) : NaNs were produced" which indicates that permutation results caused by extreme outlier expression values
  that could not be fitted accordingly. This warning will not be printed out when the corresponding outlier values are extracted from the dataset.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012</p>
<p>Quint M et al. (2012). A transcriptomic hourglass in plant embryogenesis. Nature (490): 98-101.</p>
<p>M. L. Delignette-Muller, R. Pouillot, J.-B. Denis and C. Dutang (2014), fitdistrplus: help to fit of a parametric distribution to non-censored or censored data.</p>
<p>Cullen AC and Frey HC (1999) Probabilistic techniques in exposure assessment. Plenum Press, USA, pp. 81-159.</p>
<p>Evans M, Hastings N and Peacock B (2000) Statistical distributions. John Wiley and Sons Inc.</p>
<p>Sokal RR and Rohlf FJ (1995) Biometry. W.H. Freeman and Company, USA, pp. 111-115.</p>
<p>Juergen Gross and bug fixes by Uwe Ligges (2012). nortest: Tests for Normality. R package version 1.0-2.</p>
<p>http://CRAN.R-project.org/package=nortest</p>
<p>Dallal, G.E. and Wilkinson, L. (1986): An analytic approximation to the distribution of Lilliefors test for normality. The American Statistician, 40, 294-296.</p>
<p>Stephens, M.A. (1974): EDF statistics for goodness of fit and some comparisons. Journal of the American Statistical Association, 69, 730-737.</p>
<p>http://stackoverflow.com/questions/4290081/fitting-data-to-distributions?rq=1</p>
<p>http://stats.stackexchange.com/questions/45033/can-i-use-kolmogorov-smirnov-test-and-estimate-distribution-parameters</p>
<p>http://cran.r-project.org/doc/contrib/Ricci-distributions-en.pdf</p>
<p>http://cran.r-project.org/doc/contrib/Ricci-distributions-en.pdf</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="TAI.html">TAI</a></code>, <code><a href="TDI.html">TDI</a></code>, <code><a href="PlotSignature.html">PlotSignature</a></code>, <code><a href="bootMatrix.html">bootMatrix</a></code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># read standard phylotranscriptomics data</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># example PhyloExpressionSet using 100 permutations</span></span></span>
<span class="r-in"><span><span class="fu">FlatLineTest</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>             permutations  <span class="op">=</span> <span class="fl">100</span>,</span></span>
<span class="r-in"><span>             plotHistogram <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [ Number of Eigen threads that are employed on your machine: 12 ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [ Computing age assignment permutations for test statistic ... ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [=========================================] 100%   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [ Computing variances of permuted transcriptome signatures ... ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Total runtime of your permutation test: 0.017  seconds.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -&gt; We recommended using at least 20000 permutations to achieve a sufficient permutation test.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $p.value</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1.832554e-14</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $std.dev</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.05747213 0.05645721 0.05603895 0.05508510 0.05295585 0.05042765 0.05814981</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $ks.test</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Exact one-sample Kolmogorov-Smirnov test</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> data:  filtered_vars</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> D = 0.077499, p-value = 0.6374</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> alternative hypothesis: two-sided</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># use your own permutation matrix based on which p-values (FlatLineTest)</span></span></span>
<span class="r-in"><span><span class="co"># shall be computed</span></span></span>
<span class="r-in"><span><span class="va">custom_perm_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="bootMatrix.html">bootMatrix</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>,<span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [ Number of Eigen threads that are employed on your machine: 12 ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [ Computing age assignment permutations for test statistic ... ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [=========================================] 100%   </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [ Computing variances of permuted transcriptome signatures ... ]</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">FlatLineTest</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>             custom.perm.matrix <span class="op">=</span> <span class="va">custom_perm_matrix</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -&gt; We recommended using at least 20000 permutations to achieve a sufficient permutation test.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $p.value</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 1.096419e-09</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $std.dev</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     Zygote   Quadrant   Globular      Heart    Torpedo       Bent     Mature </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 0.05054628 0.04901842 0.05031856 0.04952507 0.05249808 0.05118999 0.05082478 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $ks.test</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 	Asymptotic one-sample Kolmogorov-Smirnov test</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> data:  filtered_vars</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> D = 0.059129, p-value = 0.8756</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> alternative hypothesis: two-sided</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Nikola Kalábová, Kristian K Ullrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

