<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Perform Reductive Late Conservation Test — LateConservationTest • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Perform Reductive Late Conservation Test — LateConservationTest"><meta property="og:description" content="The Reductive Late Conservation Test aims to statistically evaluate the
existence of a monotonically decreasing phylotranscriptomic pattern based on TAI or TDI computations.
The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern) 
does not follow an late conservation like pattern. A p-value &amp;lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does
indeed follow an late conservation (high-high-low) shape."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI package</a>
    </li>
    <li>
      <a href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/myTAI/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Perform Reductive Late Conservation Test</h1>
    <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/LateConservationTest.R" class="external-link"><code>R/LateConservationTest.R</code></a></small>
    <div class="hidden name"><code>LateConservationTest.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>The <em>Reductive Late Conservation Test</em> aims to statistically evaluate the
existence of a monotonically decreasing phylotranscriptomic pattern based on <code><a href="TAI.html">TAI</a></code> or <code><a href="TDI.html">TDI</a></code> computations.
The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern) 
does not follow an late conservation like pattern. A p-value &lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does
indeed follow an late conservation (high-high-low) shape.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">LateConservationTest</span><span class="op">(</span></span>
<span>  <span class="va">ExpressionSet</span>,</span>
<span>  modules <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  permutations <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  lillie.test <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  plotHistogram <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  runs <span class="op">=</span> <span class="fl">10</span>,</span>
<span>  parallel <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  gof.warning <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  custom.perm.matrix <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ExpressionSet</dt>
<dd><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></dd>


<dt>modules</dt>
<dd><p>a list storing three elements: early, mid, and late. Each element expects a numeric
vector specifying the developmental stages or experiments that correspond to each module. 
For example, <code>module</code> = list(early = 1:2, mid = 3:5, late = 6:7) divides a dataset 
storing seven developmental stages into 3 modules.</p></dd>


<dt>permutations</dt>
<dd><p>a numeric value specifying the number of permutations to be performed for the <code>ReductiveHourglassTest</code>.</p></dd>


<dt>lillie.test</dt>
<dd><p>a boolean value specifying whether the Lilliefors Kolmogorov-Smirnov Test shall be performed to quantify the goodness of fit.</p></dd>


<dt>plotHistogram</dt>
<dd><p>a boolean value specifying whether a <em>Lillifor's Kolmogorov-Smirnov-Test</em> 
shall be performed to test the goodness of fit of the approximated distribution, as well as additional plots quantifying the significance
of the observed phylotranscriptomic pattern.</p></dd>


<dt>runs</dt>
<dd><p>specify the number of runs to be performed for goodness of fit computations, in case <code>plotHistogram</code> = <code>TRUE</code>.
In most cases <code>runs</code> = 100 is a reasonable choice. Default is <code>runs</code> = 10 (because it takes less computation time for demonstration purposes).</p></dd>


<dt>parallel</dt>
<dd><p>performing <code>runs</code> in parallel (takes all cores of your multicore machine).</p></dd>


<dt>gof.warning</dt>
<dd><p>a logical value indicating whether non significant goodness of fit results should be printed as warning. Default is <code>gof.warning = FALSE</code>.</p></dd>


<dt>custom.perm.matrix</dt>
<dd><p>a custom <code><a href="bootMatrix.html">bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic. Default is <code>custom.perm.matrix = NULL</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a list object containing the list elements:</p>


<p><code>p.value</code> : the p-value quantifying the statistical significance (low-high-high pattern) of the given phylotranscriptomics pattern.</p>


<p><code>std.dev</code> : the standard deviation of the N sampled phylotranscriptomics patterns for each developmental stage S.</p>


<p><code>lillie.test</code> : a boolean value specifying whether the <em>Lillifors KS-Test</em> returned a p-value &gt; 0.05, 
which indicates that fitting the permuted scores with a normal distribution seems plausible.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>The <em>reductive late conservation test</em> is a permutation test based on the following test statistic.</p>
<p>(1) A set of developmental stages is partitioned into three modules - early, mid, and late - based on prior biological knowledge.</p>
<p>(2) The mean <code><a href="TAI.html">TAI</a></code> or <code><a href="TDI.html">TDI</a></code> value for each of the three modules T_early, T_mid, and T_late are computed.</p>
<p>(3) The two differences D1 = T_early - T_late and D2 = T_mid - T_late are calculated.</p>
<p>(4) The minimum D_min of D1 and D2 is computed as final test statistic of the reductive hourglass test.</p>

<p>In order to determine the statistical significance of an observed minimum difference D_min 
the following permutation test was performed. Based on the <code><a href="bootMatrix.html">bootMatrix</a></code> D_min 
is calculated from each of the permuted <code><a href="TAI.html">TAI</a></code> or <code><a href="TDI.html">TDI</a></code> profiles, 
approximated by a Gaussian distribution with method of moments estimated parameters returned by <code><a href="https://rdrr.io/pkg/fitdistrplus/man/fitdist.html" class="external-link">fitdist</a></code>, 
and the corresponding p-value is computed by <code><a href="https://rdrr.io/r/stats/Normal.html" class="external-link">pnorm</a></code> given the estimated parameters of the Gaussian distribution. 
The <em>goodness of fit</em> for the random vector <em>D_min</em> is statistically quantified by an Lilliefors (Kolmogorov-Smirnov) test 
for normality.</p>

<p>In case the parameter <em>plotHistogram = TRUE</em>, a multi-plot is generated showing:</p>       
<p>(1) A Cullen and Frey skewness-kurtosis plot generated by <code><a href="https://rdrr.io/pkg/fitdistrplus/man/descdist.html" class="external-link">descdist</a></code>. 
This plot illustrates which distributions seem plausible to fit the resulting permutation vector D_min. 
In the case of the <em>reductive late conservation test</em> a normal distribution seemed plausible.</p>
<p>(2) A histogram of D_min combined with the density plot is plotted. D_min is then fitted by a normal distribution. 
The corresponding parameters are estimated by <em>moment matching estimation</em> using the <code><a href="https://rdrr.io/pkg/fitdistrplus/man/fitdist.html" class="external-link">fitdist</a></code> function.</p>
<p>(3) A plot showing the p-values for N independent runs to verify that a specific p-value is biased by a specific permutation order.</p>
<p>(4) A barplot showing the number of cases in which the underlying goodness of fit (returned by Lilliefors (Kolmogorov-Smirnov) test 
for normality) has shown to be significant (<code>TRUE</code>) or not significant (<code>FALSE</code>). 
This allows to quantify the permutation bias and their implications on the goodness of fit.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012</p>
<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.</p>
<p>Piasecka B, Lichocki P, Moretti S, et al. (2013) <em>The hourglass and the early conservation models co-existing
patterns of developmental constraints in vertebrates</em>. PLoS Genet. 9(4): e1003476.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="lcScore.html">lcScore</a></code>, <code><a href="bootMatrix.html">bootMatrix</a></code>, <code><a href="FlatLineTest.html">FlatLineTest</a></code>,<code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>, <code><a href="ReverseHourglassTest.html">ReverseHourglassTest</a></code>, <code><a href="PlotSignature.html">PlotSignature</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Hajk-Georg Drost and Jaruwatana Sodai Lotharukpong</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># perform the late conservation test for a PhyloExpressionSet</span></span></span>
<span class="r-in"><span><span class="co"># here the prior biological knowledge is that stages 1-2 correspond to module 1 = early,</span></span></span>
<span class="r-in"><span><span class="co"># stages 3-5 to module 2 = mid (phylotypic module), and stages 6-7 correspond to</span></span></span>
<span class="r-in"><span><span class="co"># module 3 = late</span></span></span>
<span class="r-in"><span><span class="fu">LateConservationTest</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>                       modules <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>early <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, mid <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">5</span>, late <span class="op">=</span> <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>                       permutations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The phylotranscriptomic pattern may not follow a late conservation pattern (high-mid-low or high-high-low).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $p.value</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.9999993</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $std.dev</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.05554113 0.05456920 0.05346371 0.05175899 0.04979477 0.05076939 0.05695157</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $lillie.test</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># use your own permutation matrix based on which p-values (LateConservationTest)</span></span></span>
<span class="r-in"><span><span class="co"># shall be computed</span></span></span>
<span class="r-in"><span><span class="va">custom_perm_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="bootMatrix.html">bootMatrix</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>,<span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">LateConservationTest</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>                       modules <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>early <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, mid <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">5</span>, late <span class="op">=</span> <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>                       custom.perm.matrix <span class="op">=</span> <span class="va">custom_perm_matrix</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The phylotranscriptomic pattern may not follow a late conservation pattern (high-mid-low or high-high-low).</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $p.value</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] 0.9999989</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $std.dev</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     Zygote   Quadrant   Globular      Heart    Torpedo       Bent     Mature </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 0.04771434 0.04892615 0.04854881 0.04874797 0.04902014 0.04889337 0.05639430 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $lillie.test</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] NA</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span>                       </span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Kristian K Ullrich, Jaruwatana Sodai Lotharukpong, Nikola Kalábová.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

