<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function matches a Phylostratigraphic Map or Divergence Map only storing unique gene ids with a ExpressionMatrix
also storing only unique gene ids."><title>Match a Phylostratigraphic Map or Divergence Map with a ExpressionMatrix — MatchMap • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Match a Phylostratigraphic Map or Divergence Map with a ExpressionMatrix — MatchMap"><meta property="og:description" content="This function matches a Phylostratigraphic Map or Divergence Map only storing unique gene ids with a ExpressionMatrix
also storing only unique gene ids."><meta property="og:image" content="https://drostlab.github.io/myTAI/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light" data-bs-theme="light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    <a class="dropdown-item" href="../articles/Enrichment.html">Enrichment Analyses</a>
    <a class="dropdown-item" href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    <a class="dropdown-item" href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    <a class="dropdown-item" href="../articles/Introduction.html">Introduction to the myTAI package</a>
    <a class="dropdown-item" href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Match a Phylostratigraphic Map or Divergence Map with a ExpressionMatrix</h1>
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/MatchMap.R" class="external-link"><code>R/MatchMap.R</code></a></small>
      <div class="d-none name"><code>MatchMap.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function matches a <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> only storing unique gene ids with a ExpressionMatrix
also storing only unique gene ids.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">MatchMap</span><span class="op">(</span><span class="va">Map</span>, <span class="va">ExpressionMatrix</span>, remove.duplicates <span class="op">=</span> <span class="cn">FALSE</span>, accumulate <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>Map</dt>
<dd><p>a standard <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> object.</p></dd>


<dt>ExpressionMatrix</dt>
<dd><p>a standard ExpressionMatrix object.</p></dd>


<dt>remove.duplicates</dt>
<dd><p>a logical value indicating whether duplicate gene ids should be removed from the data set.</p></dd>


<dt>accumulate</dt>
<dd><p>an accumulation function such as <code><a href="https://rdrr.io/r/base/mean.html" class="external-link">mean()</a></code>, <code><a href="https://rdrr.io/r/stats/median.html" class="external-link">median()</a></code>, or <code><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min()</a></code>
to accumulate multiple expression levels that map to the same unique gene id present in the <code>ExpressionMatrix</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>In phylotranscriptomics analyses two major techniques are performed to
obtain standard <em>Phylostratigraphic map</em> or <em>Divergence map</em> objects.</p>
<p>To obtain a <em>Phylostratigraphic Map</em>, <em>Phylostratigraphy</em> (Domazet-Loso et al., 2007) has to be performed. To obtain a <em>Divergence Map</em>, 
orthologous gene detection, Ka/Ks computations, and decilation (Quint et al., 2012; Drost et al., 2015) have to be performed.</p>
<p>The resulting standard <em>Phylostratigraphic Map</em> or <em>Divergence Map</em> objects consist of 2 colums storing the phylostratum assignment 
or divergence stratum assignment of a given gene in column one, and the corresponding gene id of that gene on columns two.</p>
<p>A standard ExpressionMatrix is a common gene expression matrix storing the gene ids or probe ids in the first column, and all experiments/stages/replicates in the following columns.</p>
<p>The <em>MatchMap</em> function takes both standard datasets: <em>Map</em> and <em>ExpressionMatrix</em> to merge both data sets to obtain a standard PhyloExpressionSet or DivergenceExpressionSet object.</p>
<p>This procedure is analogous to <code><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></code>, but is customized to the <em>Phylostratigraphic Map</em>, <em>Divergence Map</em>, and <em>ExpressionMatrix</em> standards to allow a faster and more intuitive usage.</p>
<p>In case you work with an ExpressionMatrix that stores multiple expression levels for a unique gene id, you
can specify the <code>accumulation</code> argument to accumulate these multiple expression levels to obtain
one expression level for one unique gene.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Domazet-Loso T, Brajkovic J, Tautz D (2007) A phylostratigraphy approach to uncover the genomic history of major adaptations in metazoan lineages. Trends Genet. 23: 533-9.</p>
<p>Domazet-Loso T, Tautz D (2010) A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns. Nature 468: 815-8.</p>
<p>Quint M., Drost H.G., Gabel A., Ullrich K.K., Boenn M., Grosse I. (2012) A transcriptomic hourglass in plant embryogenesis. Nature 490: 98-101.</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span>        </span></span>
<span class="r-in"><span><span class="co"># load a standard PhyloExpressionSet</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span>        </span></span>
<span class="r-in"><span><span class="co"># in a standard PhyloExpressionSet, </span></span></span>
<span class="r-in"><span><span class="co"># column one and column two denote a standard </span></span></span>
<span class="r-in"><span><span class="co"># phylostratigraphic map</span></span></span>
<span class="r-in"><span><span class="va">PhyloMap</span> <span class="op">&lt;-</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span> , <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></span>
<span class="r-in"><span>        </span></span>
<span class="r-in"><span><span class="co"># look at the phylostratigraphic map standard</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">PhyloMap</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Phylostratum      GeneID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1 at1g01040.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1 at1g01050.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            1 at1g01070.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            1 at1g01080.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            1 at1g01090.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6            1 at1g01120.1</span>
<span class="r-in"><span>        </span></span>
<span class="r-in"><span><span class="co"># in a standard PhyloExpressionSet, column two combined </span></span></span>
<span class="r-in"><span><span class="co"># with column 3 - N denote a standard ExpressionMatrix</span></span></span>
<span class="r-in"><span><span class="va">ExpressionMatrixExample</span> <span class="op">&lt;-</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span> , <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">2</span>,<span class="fl">3</span><span class="op">:</span><span class="fl">9</span><span class="op">)</span><span class="op">]</span></span></span>
<span class="r-in"><span>        </span></span>
<span class="r-in"><span><span class="co"># these two data sets shall illustrate an example </span></span></span>
<span class="r-in"><span><span class="co"># phylostratigraphic map that is returned</span></span></span>
<span class="r-in"><span><span class="co"># by a standard phylostratigraphy run, and a expression set </span></span></span>
<span class="r-in"><span><span class="co"># that is the result of expression data analysis </span></span></span>
<span class="r-in"><span><span class="co"># (background correction, normalization, ...)</span></span></span>
<span class="r-in"><span>        </span></span>
<span class="r-in"><span><span class="co"># now we can use the MatchMap function to merge both data sets</span></span></span>
<span class="r-in"><span><span class="co"># to obtain a standard PhyloExpressionSet</span></span></span>
<span class="r-in"><span>        </span></span>
<span class="r-in"><span><span class="va">PES</span> <span class="op">&lt;-</span> <span class="fu">MatchMap</span><span class="op">(</span><span class="va">PhyloMap</span>, <span class="va">ExpressionMatrixExample</span><span class="op">)</span></span></span>
<span class="r-in"><span>        </span></span>
<span class="r-in"><span><span class="co"># note that the function returns a head() </span></span></span>
<span class="r-in"><span><span class="co"># of the matched gene ids to enable</span></span></span>
<span class="r-in"><span><span class="co"># the user to find potential mis-matches</span></span></span>
<span class="r-in"><span>        </span></span>
<span class="r-in"><span><span class="co"># the entire procedure is analogous to merge() </span></span></span>
<span class="r-in"><span><span class="co"># with two data sets sharing the same gene ids </span></span></span>
<span class="r-in"><span><span class="co"># as column (primary key)</span></span></span>
<span class="r-in"><span><span class="va">PES_merge</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/merge.html" class="external-link">merge</a></span><span class="op">(</span><span class="va">PhyloMap</span>, <span class="va">ExpressionMatrixExample</span><span class="op">)</span></span></span>
<span class="r-in"><span>        </span></span>
<span class="r-in"><span>        </span></span>
<span class="r-in"><span>        </span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Nikola Kalábová, Kristian K Ullrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

