<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Plot evolutionary signatures across transcriptomes, removing top level genes — PlotSignatureGeneQuantiles • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Plot evolutionary signatures across transcriptomes, removing top level genes — PlotSignatureGeneQuantiles"><meta name="description" content="PlotSignatureGeneQuantiles is used to investigate
the robustness of a transcriptomic index signal to removing the most highly expressed genes
from the given expression set. The resulting perturbed signal is plotted for
different quantile probability thresholds defining the set of top level genes."><meta property="og:description" content="PlotSignatureGeneQuantiles is used to investigate
the robustness of a transcriptomic index signal to removing the most highly expressed genes
from the given expression set. The resulting perturbed signal is plotted for
different quantile probability thresholds defining the set of top level genes."><meta property="og:image" content="https://drostlab.github.io/myTAI/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/myTAI.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Tutorials</h6></li>
    <li><a class="dropdown-item" href="../articles/Introduction.html">Introduction</a></li>
    <li><a class="dropdown-item" href="../articles/Intermediate.html">Intermediate</a></li>
    <li><a class="dropdown-item" href="../articles/Advanced.html">Advanced</a></li>
    <li><hr class="dropdown-divider"></li>
    <li><h6 class="dropdown-header" data-toc-skip>Discussion</h6></li>
    <li><a class="dropdown-item" href="../articles/Phylostratigraphy.html">Genomic Phylostratigraphy</a></li>
    <li><a class="dropdown-item" href="../articles/Enrichment.html">Enrichment Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Expression.html">Gene Expression Analysis</a></li>
    <li><a class="dropdown-item" href="../articles/Taxonomy.html">Taxonomic Retrieval</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Plot evolutionary signatures across transcriptomes, removing top level genes</h1>
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/PlotSignatureGeneQuantiles.R" class="external-link"><code>R/PlotSignatureGeneQuantiles.R</code></a></small>
      <div class="d-none name"><code>PlotSignatureGeneQuantiles.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p><em>PlotSignatureGeneQuantiles</em> is used to investigate
the robustness of a transcriptomic index signal to removing the most highly expressed genes
from the given expression set. The resulting perturbed signal is plotted for
different quantile probability thresholds defining the set of top level genes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">PlotSignatureGeneQuantiles</span><span class="op">(</span></span>
<span>  <span class="va">ExpressionSet</span>,</span>
<span>  quantiles <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0.99</span>, <span class="fl">0.95</span>, <span class="fl">0.9</span>, <span class="fl">0.8</span><span class="op">)</span>,</span>
<span>  gene.selection.criterion <span class="op">=</span> <span class="st">"mean"</span>,</span>
<span>  measure <span class="op">=</span> <span class="st">"TAI"</span>,</span>
<span>  TestStatistic <span class="op">=</span> <span class="st">"FlatLineTest"</span>,</span>
<span>  modules <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  permutations <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  p.value <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  shaded.area <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Ontogeny"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Transcriptome Index"</span>,</span>
<span>  main <span class="op">=</span> <span class="st">""</span>,</span>
<span>  lwd <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  y.ticks <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-expressionset">ExpressionSet<a class="anchor" aria-label="anchor" href="#arg-expressionset"></a></dt>
<dd><p>a standard PhyloExpressionSet, DivergenceExpressionSet or PolymorphismsExpressionSet object.</p></dd>


<dt id="arg-quantiles">quantiles<a class="anchor" aria-label="anchor" href="#arg-quantiles"></a></dt>
<dd><p>a numeric vector of quantile probabilities (between [0,1])</p></dd>


<dt id="arg-gene-selection-criterion">gene.selection.criterion<a class="anchor" aria-label="anchor" href="#arg-gene-selection-criterion"></a></dt>
<dd><p>a string defining the criterion by which genes should be selected
Possible values can be:</p><ul><li><p><code>gene.selection.criterion</code> = <code>"mean"</code> : Sort the genes by the mean of their gene expression (equivalently, by their total expression counts across stages)</p></li>
<li><p><code>gene.selection.criterion</code> = <code>"variance"</code> : Sort the genes by the variance of their gene expression across stages</p></li>
</ul></dd>


<dt id="arg-measure">measure<a class="anchor" aria-label="anchor" href="#arg-measure"></a></dt>
<dd><p>type of transcriptome index that shall be computed. E.g.</p><ul><li><p><code>measure = "TAI"</code> (Transcriptome Age Index)</p></li>
<li><p><code>measure = "TDI"</code> (Transcriptome Divergence Index)</p></li>
<li><p><code>measure = "TPI"</code> (Transcriptome Polymorphism Index)</p></li>
</ul></dd>


<dt id="arg-teststatistic">TestStatistic<a class="anchor" aria-label="anchor" href="#arg-teststatistic"></a></dt>
<dd><p>a string defining the type of test statistics to be used to quantify the statistical significance the present phylotranscriptomics pattern.
Possible values can be:</p><ul><li><p><code>TestStatistic</code> = <code>"FlatLineTest"</code> : Statistical test for the deviation from a flat line</p></li>
<li><p><code>TestStatistic</code> = <code>"ReductiveHourglassTest"</code> : Statistical test for the existence of a hourglass shape (high-low-high pattern)</p></li>
<li><p><code>TestStatistic</code> = <code>"ReverseHourglassTest"</code> : Statistical test for the existence of a reverse hourglass pattern (low-high-low pattern)</p></li>
<li><p><code>TestStatistic</code> = <code>"EarlyConservationTest"</code> : Statistical test for the existence of a early conservation pattern (low-high-high pattern)</p></li>
<li><p><code>TestStatistic</code> = <code>"LateConservationTest"</code> : Statistical test for the existence of a late conservation pattern (high-high-low pattern)</p></li>
</ul></dd>


<dt id="arg-modules">modules<a class="anchor" aria-label="anchor" href="#arg-modules"></a></dt>
<dd><p>a list storing three elements for the <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>, <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>, <code><a href="LateConservationTest.html">LateConservationTest</a></code>,
or <code><a href="ReverseHourglassTest.html">ReverseHourglassTest</a></code>: early, mid, and late.
Each element expects a numeric vector specifying the developmental stages
or experiments that correspond to each module. For example:</p><ul><li><p><code>modules</code> = <code>list(early = 1:2, mid = 3:5, late = 6:7)</code> divides a dataset storing seven developmental stages into 3 modules.</p></li>
</ul></dd>


<dt id="arg-permutations">permutations<a class="anchor" aria-label="anchor" href="#arg-permutations"></a></dt>
<dd><p>a numeric value specifying the number of permutations to be performed for the <code><a href="FlatLineTest.html">FlatLineTest</a></code>, <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>, <code><a href="LateConservationTest.html">LateConservationTest</a></code>, <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code> or <code><a href="ReverseHourglassTest.html">ReverseHourglassTest</a></code>.</p></dd>


<dt id="arg-p-value">p.value<a class="anchor" aria-label="anchor" href="#arg-p-value"></a></dt>
<dd><p>a boolean value specifying whether the p-value of the test statistic shall be printed within the legend, for each expression set.</p></dd>


<dt id="arg-shaded-area">shaded.area<a class="anchor" aria-label="anchor" href="#arg-shaded-area"></a></dt>
<dd><p>a boolean value specifying whether a shaded area shall
be drawn for the developmental stages defined to be the presumptive phylotypic period.</p></dd>


<dt id="arg-xlab">xlab<a class="anchor" aria-label="anchor" href="#arg-xlab"></a></dt>
<dd><p>label of x-axis.</p></dd>


<dt id="arg-ylab">ylab<a class="anchor" aria-label="anchor" href="#arg-ylab"></a></dt>
<dd><p>label of y-axis.</p></dd>


<dt id="arg-main">main<a class="anchor" aria-label="anchor" href="#arg-main"></a></dt>
<dd><p>figure title.</p></dd>


<dt id="arg-lwd">lwd<a class="anchor" aria-label="anchor" href="#arg-lwd"></a></dt>
<dd><p>line width.</p></dd>


<dt id="arg-alpha">alpha<a class="anchor" aria-label="anchor" href="#arg-alpha"></a></dt>
<dd><p>transparency of the shaded area and error ribbon (between [0,1]). Default is <code>alpha = 0.1</code>.</p></dd>


<dt id="arg-y-ticks">y.ticks<a class="anchor" aria-label="anchor" href="#arg-y-ticks"></a></dt>
<dd><p>number of ticks on the y-axis. Default is <code>ticks = 3</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>a ggplot object visualising the transcriptome index of the
expression set, together with its standard deviation per stage,
obtained by permuting the gene ages.
For each quantile probability threshold, the resulting perturbed signal is shown
as a separate profile.
The profiles are shown on the same axes, so that they can be readily compared.
Optionally, the p-value of each profile, with respect to the choice of statistic,
is shown.</p>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Stefan Manolache</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span>                      </span></span>
<span class="r-in"><span><span class="co"># Flat line test, select top expressed genes</span></span></span>
<span class="r-in"><span><span class="fu">PlotSignatureGeneQuantiles</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>                           quantiles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1.0</span>, <span class="fl">0.99</span>, <span class="fl">0.95</span>, <span class="fl">0.90</span>, <span class="fl">0.80</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                           main<span class="op">=</span><span class="st">"Excluding top level genes by total </span></span></span>
<span class="r-in"><span><span class="st">                           expression using different thresholds"</span>,</span></span>
<span class="r-in"><span>                           gene.selection.criterion<span class="op">=</span><span class="st">"mean"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in PlotSignatureGeneQuantiles(ExpressionSet = PhyloExpressionSetExample,     quantiles = c(1, 0.99, 0.95, 0.9, 0.8), main = "Excluding top level genes by total \n                           expression using different thresholds",     gene.selection.criterion = "mean"):</span> could not find function "PlotSignatureGeneQuantiles"</span>
<span class="r-in"><span><span class="co"># Flat line test, select top genes by variance of expression</span></span></span>
<span class="r-in"><span><span class="fu">PlotSignatureGeneQuantiles</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>                           main<span class="op">=</span><span class="st">"Excluding top level genes by variance of </span></span></span>
<span class="r-in"><span><span class="st">                           expression using different thresholds"</span>,</span></span>
<span class="r-in"><span>                           quantiles<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1.0</span>, <span class="fl">0.99</span>, <span class="fl">0.95</span>, <span class="fl">0.90</span>, <span class="fl">0.80</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                           gene.selection.criterion<span class="op">=</span><span class="st">"variance"</span><span class="op">)</span></span></span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in PlotSignatureGeneQuantiles(ExpressionSet = PhyloExpressionSetExample,     main = "Excluding top level genes by variance of \n                           expression using different thresholds",     quantiles = c(1, 0.99, 0.95, 0.9, 0.8), gene.selection.criterion = "variance"):</span> could not find function "PlotSignatureGeneQuantiles"</span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich, Stefan Manolache.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

