<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot the Transcriptome Age Index or Transcriptome Divergence Index — PlotPattern • myTAI</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot the Transcriptome Age Index or Transcriptome Divergence Index — PlotPattern"><meta property="og:description" content="Function to plot the TAI or TDI of a given PhyloExpressionSet or DivergenceExpressionSet object.
This function plot the TAI or TDI of a given PhyloExpressionSet or DivergenceExpressionSet object."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1.0000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI package</a>
    </li>
    <li>
      <a href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/myTAI/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot the Transcriptome Age Index or Transcriptome Divergence Index</h1>
    <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/PlotPattern.R" class="external-link"><code>R/PlotPattern.R</code></a></small>
    <div class="hidden name"><code>PlotPattern.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Function to plot the TAI or TDI of a given PhyloExpressionSet or DivergenceExpressionSet object.
This function plot the <code><a href="TAI.html">TAI</a></code> or <code><a href="TDI.html">TDI</a></code> of a given PhyloExpressionSet or DivergenceExpressionSet object.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">PlotPattern</span><span class="op">(</span></span>
<span>  <span class="va">ExpressionSet</span>,</span>
<span>  TestStatistic <span class="op">=</span> <span class="st">"FlatLineTest"</span>,</span>
<span>  modules <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  permutations <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  lillie.test <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  digits.ylab <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  p.value <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  shaded.area <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  y.ticks <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  custom.perm.matrix <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ExpressionSet</dt>
<dd><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></dd>


<dt>TestStatistic</dt>
<dd><p>a string defining the type of test statistics to be used to quantify the statistical significance the present phylotranscriptomics pattern.
Possible values can be: <code>TestStatistic</code> = <code>"FlatLineTest"</code> : Statistical test for the deviation from a flat line.
<code>TestStatistic</code> = <code>"ReductiveHourglassTest"</code> : Statistical test for the existence of a hourglass shape (high-low-high pattern).
<code>TestStatistic</code> = <code>"EarlyConservationTest"</code> : Statistical test for the existence of a earlyconservation pattern (low-high-high pattern).</p></dd>


<dt>modules</dt>
<dd><p>a list storing three elements for the <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code> or <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>: early, mid, and late. 
Each element expects a numeric vector specifying the developmental stages 
or experiments that correspond to each module. For example, 
<code>module</code> = <code>list(early = 1:2, mid = 3:5, late = 6:7)</code> devides a dataset storing seven developmental stages into 3 modules.</p></dd>


<dt>permutations</dt>
<dd><p>a numeric value specifying the number of permutations to be performed for the <code><a href="FlatLineTest.html">FlatLineTest</a></code>, <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code> or <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>.</p></dd>


<dt>lillie.test</dt>
<dd><p>a boolean value specifying whether the Lilliefors Kolmogorov-Smirnov Test shall be performed.</p></dd>


<dt>digits.ylab</dt>
<dd><p>a numeric value specifying the number of digits shown for the TAI or TDI values on the y-axis.</p></dd>


<dt>p.value</dt>
<dd><p>a boolean value specifying whether the p-value of the test statistic shall be printed within the plot area.</p></dd>


<dt>shaded.area</dt>
<dd><p>a boolean value specifying whether a shaded area shall 
be drawn for the developmental stages defined to be the presumptive phylotypic period.</p></dd>


<dt>y.ticks</dt>
<dd><p>a numeric value specifying the number of ticks to be drawn on the y-axis.</p></dd>


<dt>custom.perm.matrix</dt>
<dd><p>a custom <code><a href="bootMatrix.html">bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic visualized by <code>PlotPattern</code>. Default is <code>custom.perm.matrix = NULL</code>.</p></dd>


<dt>...</dt>
<dd><p>default plot parameters.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a plot visualizing the phylotranscriptomic pattern of a given PhyloExpressionSet or DivergenceExpressionSet object.</p>


<p>The corresponding <em>p-value</em> of the test statistic is named as follows:</p>


<p><code>p_flt</code> = p-value of the corresponding <code><a href="FlatLineTest.html">FlatLineTest</a></code></p>



<p><code>p_rht</code> = p-value of the corresponding <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code></p>



<p><code>p_ect</code> = p-value of the corresponding <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code></p>


    </div>
    <div id="details">
    <h2>Details</h2>
    <p>This function computes a permutation test quantifying the statistical significance of the prensent phylotranscriptomics pattern. 
The user can choose between the <code><a href="FlatLineTest.html">FlatLineTest</a></code>, <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>, or <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>. 
The <code><a href="FlatLineTest.html">FlatLineTest</a></code> tests for any significant deviation from a flat line. 
Each period or stage that significantly deviates from a flat line, might be governed by stronger selective pressure (in terms of natural selection) compared to other stages or periods of development.
The <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code> specificly tests for the statistical significance of a molecular hourglass pattern (high-low-high pattern) with prior biological knowlegde.
The corresponding p-value that is printed within the plot (by default) specifies the statistical significance of the chosen test statistic.</p>
<p>The <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code> specificly tests for the statistical significance of a low-high-high pattern (monotonically increasing function)
with prior biological knowlegde.
The corresponding p-value that is printed within the plot (by default) specifies the statistical significance of the chosen test statistic.</p>
<p>The x-axis denotes the developmental series (time course / experiments / ontogeny) of the input ExpressionSet and the y-axis 
denotes the corresponding mean transcriptome age value (<code><a href="TAI.html">TAI</a></code> or <code><a href="TDI.html">TDI</a></code>) of the given ExpressionSet.</p>
<p>Furthermore, the grey lines above and below the actual phylotranscriptomics pattern denotes the standard deviations of <code><a href="TAI.html">TAI</a></code> or <code><a href="TDI.html">TDI</a></code> 
values that have been estimated from the <code><a href="bootMatrix.html">bootMatrix</a></code>.
A low mean transcriptome age value denotes an evolutionary older transcriptome being active during the corresponding periods, 
whereas a high mean transcriptome age value denotes an evolutionary younger transcriptome being active during the corresponding periods.
For mean transcriptome divergence, a low mean transcriptome divergence value denotes a more conserved transcriptome 
being active (between two species), whereas a high mean transcriptome divergence value denotes a more divergent transcriptome 
being active (between two species) - in terms of protein-sequence substitution rates.</p>
<p>This function is useful to fastly plot the <code><a href="TAI.html">TAI</a></code> or <code><a href="TDI.html">TDI</a></code> 
profile of a given PhyloExpressionSet or DivergenceExpressionSet object and 
the statistical significance of the corresponding pattern.
Internally the function calls several graphics functions, 
such as <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></code>, <code><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></code>, and <code><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></code>. 
For the ellipsis argument <code><a href="https://rdrr.io/r/base/dots.html" class="external-link">...</a></code> all graphics specific arguments can be defined. 
Internally the function specific arguments for e.g. <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></code>, <code><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></code>, 
and <code><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></code> will be detected and are passed to the corresponding function.</p>
<p>Hence, when calling the function <code>PlotPattern</code>, one can specify arguments 
for <code><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></code> and <code><a href="https://rdrr.io/r/graphics/axis.html" class="external-link">axis</a></code> and <code><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></code> 
as <code><a href="https://rdrr.io/r/base/dots.html" class="external-link">...</a></code>.</p>
<p>In case prior biological knowledge is present for a specific period of development,
the <code>shaded.area</code> argument can be set to <code>TRUE</code> and the function will use
the values stored in the <code>mid</code> argument to draw a shaded area within the corresponding period of development.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Domazet-Loso T and Tautz D. (2010). <em>A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns</em>. Nature (468): 815-818.</p>
<p>Quint M et al. (2012). <em>A transcriptomic hourglass in plant embryogenesis</em>. Nature (490): 98-101.</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="TAI.html">TAI</a></code>, <code><a href="TDI.html">TDI</a></code>, <code><a href="FlatLineTest.html">FlatLineTest</a></code>,
 <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>, <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>,
 <code><a href="PlotSignature.html">PlotSignature</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># load PhyloExpressionSet</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># only visualize the TAI profile without any test statistics...</span></span></span>
<span class="r-in"><span><span class="co"># this is equavalent to performing: plot(TAI(PhyloExpressionSetExample), type = "l", lwd = 6)</span></span></span>
<span class="r-in"><span><span class="fu">PlotPattern</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>            TestStatistic <span class="op">=</span> <span class="cn">NULL</span>,</span></span>
<span class="r-in"><span>            type          <span class="op">=</span> <span class="st">"l"</span>,</span></span>
<span class="r-in"><span>            xlab          <span class="op">=</span> <span class="st">"Ontogeny"</span>,</span></span>
<span class="r-in"><span>            ylab          <span class="op">=</span> <span class="st">"TAI"</span>,</span></span>
<span class="r-in"><span>            lwd           <span class="op">=</span> <span class="fl">9</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="PlotPattern-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># the simplest example of plotting the TAI profile of a given PhyloExpressionSet:</span></span></span>
<span class="r-in"><span><span class="co"># In this case (default) the FlatLineTest will be performed to quantify</span></span></span>
<span class="r-in"><span><span class="co"># the statistical significance of the present TAI pattern and will be drawn as 'p = ... '</span></span></span>
<span class="r-in"><span><span class="co"># in the plot</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">PlotPattern</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>            TestStatistic <span class="op">=</span> <span class="st">"FlatLineTest"</span>,</span></span>
<span class="r-in"><span>            permutations  <span class="op">=</span> <span class="fl">100</span>, </span></span>
<span class="r-in"><span>            type          <span class="op">=</span> <span class="st">"l"</span>, </span></span>
<span class="r-in"><span>            xlab          <span class="op">=</span> <span class="st">"Ontogeny"</span>, </span></span>
<span class="r-in"><span>            ylab          <span class="op">=</span> <span class="st">"TAI"</span>, </span></span>
<span class="r-in"><span>            lwd           <span class="op">=</span> <span class="fl">9</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> It is recommended to use at least 20000 permutations.</span>
<span class="r-plt img"><img src="PlotPattern-2.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># an example performing the ReductiveHourglassTest and printing the p-value</span></span></span>
<span class="r-in"><span><span class="co"># and shaded area of the presumptive phylotypic period into the plot</span></span></span>
<span class="r-in"><span><span class="co"># Here the 'p = ...' denotes the p-value that is returned by the ReductiveHourglassTest</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">PlotPattern</span><span class="op">(</span></span></span>
<span class="r-in"><span>            ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>            TestStatistic <span class="op">=</span> <span class="st">"ReductiveHourglassTest"</span>,</span></span>
<span class="r-in"><span>            modules       <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>early <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,mid <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">5</span>,late <span class="op">=</span> <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>            permutations  <span class="op">=</span> <span class="fl">100</span>, </span></span>
<span class="r-in"><span>            p.value       <span class="op">=</span> <span class="cn">TRUE</span>, </span></span>
<span class="r-in"><span>            shaded.area   <span class="op">=</span> <span class="cn">TRUE</span>, </span></span>
<span class="r-in"><span>            xlab          <span class="op">=</span> <span class="st">"Ontogeny"</span>, </span></span>
<span class="r-in"><span>            ylab          <span class="op">=</span> <span class="st">"TAI"</span>, </span></span>
<span class="r-in"><span>            type          <span class="op">=</span> <span class="st">"l"</span>, </span></span>
<span class="r-in"><span>            lwd           <span class="op">=</span> <span class="fl">9</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="PlotPattern-3.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># testing for early conservation model </span></span></span>
<span class="r-in"><span><span class="fu">PlotPattern</span><span class="op">(</span> ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>             TestStatistic <span class="op">=</span> <span class="st">"EarlyConservationTest"</span>,</span></span>
<span class="r-in"><span>            modules        <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>early <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>,mid <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">5</span>,late <span class="op">=</span> <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>            permutations   <span class="op">=</span> <span class="fl">100</span>,</span></span>
<span class="r-in"><span>            p.value        <span class="op">=</span> <span class="cn">TRUE</span>, </span></span>
<span class="r-in"><span>            shaded.area    <span class="op">=</span> <span class="cn">TRUE</span>, </span></span>
<span class="r-in"><span>            xlab           <span class="op">=</span> <span class="st">"Ontogeny"</span>, </span></span>
<span class="r-in"><span>            ylab           <span class="op">=</span> <span class="st">"TAI"</span>, </span></span>
<span class="r-in"><span>            type           <span class="op">=</span> <span class="st">"l"</span>, </span></span>
<span class="r-in"><span>            lwd            <span class="op">=</span> <span class="fl">9</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The phylotranscriptomic pattern may not follow an early conservation pattern (low-mid-high or low-high-high).</span>
<span class="r-plt img"><img src="PlotPattern-4.png" alt="" width="700" height="433"></span>
<span class="r-in"><span>            </span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># use your own permutation matrix</span></span></span>
<span class="r-in"><span><span class="va">custom_perm_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="bootMatrix.html">bootMatrix</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>,<span class="fl">100</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">PlotPattern</span><span class="op">(</span>ExpressionSet      <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>            TestStatistic      <span class="op">=</span> <span class="st">"FlatLineTest"</span>,</span></span>
<span class="r-in"><span>            custom.perm.matrix <span class="op">=</span> <span class="va">custom_perm_matrix</span>, </span></span>
<span class="r-in"><span>            type               <span class="op">=</span> <span class="st">"l"</span>, </span></span>
<span class="r-in"><span>            xlab               <span class="op">=</span> <span class="st">"Ontogeny"</span>, </span></span>
<span class="r-in"><span>            ylab               <span class="op">=</span> <span class="st">"TAI"</span>, </span></span>
<span class="r-in"><span>            lwd                <span class="op">=</span> <span class="fl">9</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> It is recommended to use at least 20000 permutations.</span>
<span class="r-plt img"><img src="PlotPattern-5.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Kristian K Ullrich.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

