<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot the Expression Profiles of a Gene Set — PlotGeneSet • myTAI</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot the Expression Profiles of a Gene Set — PlotGeneSet"><meta property="og:description" content="This function simply visualizes the gene expression profiles of
a defined subset of genes stored in the input ExpressionSet."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI package</a>
    </li>
    <li>
      <a href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/myTAI/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot the Expression Profiles of a Gene Set</h1>
    <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/PlotGeneSet.R" class="external-link"><code>R/PlotGeneSet.R</code></a></small>
    <div class="hidden name"><code>PlotGeneSet.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function simply visualizes the gene expression profiles of
a defined subset of genes stored in the input <code>ExpressionSet</code>.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">PlotGeneSet</span><span class="op">(</span></span>
<span>  <span class="va">ExpressionSet</span>,</span>
<span>  <span class="va">gene.set</span>,</span>
<span>  get.subset <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  use.only.map <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  colors <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  plot.legend <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  y.ticks <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  digits.ylab <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ExpressionSet</dt>
<dd><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></dd>


<dt>gene.set</dt>
<dd><p>a character vector storing the gene ids for which gene expression profiles shall be visualized.</p></dd>


<dt>get.subset</dt>
<dd><p>a logical value indicating whether or not an <code>ExpressionSet</code> subset of the selected <code>gene.set</code> should be retuned.</p></dd>


<dt>use.only.map</dt>
<dd><p>a logical value indicating whether instead of a standard <code>ExpressionSet</code> only a <code>Phylostratigraphic Map</code> or <code>Divergene Map</code> is passed to the function.</p></dd>


<dt>colors</dt>
<dd><p>colors for gene expression profiles. Default: <code>colors = NULL</code>, hence default colours are used.</p></dd>


<dt>plot.legend</dt>
<dd><p>a logical value indicating whether gene ids should be printed as legend next to the plot.</p></dd>


<dt>y.ticks</dt>
<dd><p>a numeric value specifying the number of ticks to be drawn on the y-axis.</p></dd>


<dt>digits.ylab</dt>
<dd><p>a numeric value specifying the number of digits shown for the expression levels on the y-axis.</p></dd>


<dt>...</dt>
<dd><p>additional parameters passed to <code><a href="https://rdrr.io/r/graphics/matplot.html" class="external-link">matplot</a></code>.</p></dd>

</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>This function simply visualizes or subsets the gene expression levels of a set of genes
that are stored in the input <code>ExpressionSet</code>.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="SelectGeneSet.html">SelectGeneSet</a></code>, <code><a href="PlotEnrichment.html">PlotEnrichment</a></code>, <code><a href="DiffGenes.html">DiffGenes</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># the best parameter setting to visualize this plot:</span></span></span>
<span class="r-in"><span><span class="co"># png("test_png.png",700,400)</span></span></span>
<span class="r-in"><span><span class="fu">PlotGeneSet</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>            gene.set      <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">2</span><span class="op">]</span>, </span></span>
<span class="r-in"><span>            type          <span class="op">=</span> <span class="st">"l"</span>, </span></span>
<span class="r-in"><span>            lty           <span class="op">=</span> <span class="fl">1</span>, </span></span>
<span class="r-in"><span>            lwd           <span class="op">=</span> <span class="fl">4</span>,</span></span>
<span class="r-in"><span>            xlab          <span class="op">=</span> <span class="st">"Ontogeny"</span>,</span></span>
<span class="r-in"><span>            ylab          <span class="op">=</span> <span class="st">"Expression Level"</span><span class="op">)</span></span></span>
<span class="r-plt img"><img src="PlotGeneSet-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># dev.off()</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># In case you would like to work with the expression levels</span></span></span>
<span class="r-in"><span><span class="co"># of selected genes you can specify the 'get.subset' argument:</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">PlotGeneSet</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>            gene.set      <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">2</span><span class="op">]</span>, </span></span>
<span class="r-in"><span>            get.subset    <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Phylostratum      GeneID    Zygote   Quadrant  Globular      Heart   Torpedo</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1 at1g01040.2  2173.635  1911.2001  1152.555  1291.4224  1000.253</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1 at1g01050.1  1501.014  1817.3086  1665.309  1564.7612  1496.321</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            1 at1g01070.1  1212.793  1233.0023   939.200   929.6195   864.218</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            1 at1g01080.2  1016.920   936.3837  1181.338  1329.4734  1392.643</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            1 at1g01090.1 11424.567 16778.1685 34366.649 39775.6405 56231.569</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         Bent    Mature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1   962.9772 1696.4274</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  1114.6435 1071.6555</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   877.2060  894.8189</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  1287.9746  861.2605</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 66980.3673 7772.5617</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># get a gene subset using only a phylostratihraphic map</span></span></span>
<span class="r-in"><span><span class="va">ExamplePSMap</span> <span class="op">&lt;-</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span> , <span class="fl">1</span><span class="op">:</span><span class="fl">2</span><span class="op">]</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu">PlotGeneSet</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">ExamplePSMap</span>, </span></span>
<span class="r-in"><span>            gene.set      <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">2</span><span class="op">]</span>, </span></span>
<span class="r-in"><span>            get.subset    <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>            use.only.map  <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Phylostratum      GeneID</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1 at1g01040.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1 at1g01050.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            1 at1g01070.1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            1 at1g01080.2</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            1 at1g01090.1</span>
<span class="r-in"><span>            </span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Kristian K Ullrich.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

