<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot evolutionary signatures across transcriptomes and RNA-seq transformations — PlotSignatureTransformed • myTAI</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot evolutionary signatures across transcriptomes and RNA-seq transformations — PlotSignatureTransformed"><meta property="og:description" content="PlotSignatureTransformed aims to statistically evaluate the
stability of ReductiveHourglassTest, FlatLineTest,
ReverseHourglassTest, EarlyConservationTest, or
LateConservationTest
(all based on TAI or TDI computations) against different
data transformations AND plot the resulting TAI profiles using PlotSignature.
The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern)
does not support the chosen test. A p-value &amp;lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does
indeed support the chosen test."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI package</a>
    </li>
    <li>
      <a href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/myTAI/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot evolutionary signatures across transcriptomes and RNA-seq transformations</h1>
    <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/PlotSignatureTransformed.R" class="external-link"><code>R/PlotSignatureTransformed.R</code></a></small>
    <div class="hidden name"><code>PlotSignatureTransformed.Rd</code></div>
    </div>

    <div class="ref-description">
    <p><em>PlotSignatureTransformed</em> aims to statistically evaluate the
stability of <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>, <code><a href="FlatLineTest.html">FlatLineTest</a></code>,
<code><a href="ReverseHourglassTest.html">ReverseHourglassTest</a></code>, <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>, or
<code><a href="LateConservationTest.html">LateConservationTest</a></code>
(all based on <code><a href="TAI.html">TAI</a></code> or <code><a href="TDI.html">TDI</a></code> computations) against different
data transformations AND plot the resulting TAI profiles using <code><a href="PlotSignature.html">PlotSignature</a></code>.
The corresponding p-value quantifies the probability that a given TAI or TDI pattern (or any phylotranscriptomics pattern)
does not support the chosen test. A p-value &lt; 0.05 indicates that the corresponding phylotranscriptomics pattern does
indeed support the chosen test.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">PlotSignatureTransformed</span><span class="op">(</span></span>
<span>  <span class="va">ExpressionSet</span>,</span>
<span>  measure <span class="op">=</span> <span class="st">"TAI"</span>,</span>
<span>  TestStatistic <span class="op">=</span> <span class="st">"FlatLineTest"</span>,</span>
<span>  transforms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"log2"</span>, <span class="st">"sqrt"</span>, <span class="st">"vst"</span>, <span class="st">"rank"</span>, <span class="st">"squared"</span><span class="op">)</span>,</span>
<span>  modules <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  permutations <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  pseudocount <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  p.value <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  shaded.area <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Ontogeny"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Transcriptome Index"</span>,</span>
<span>  main <span class="op">=</span> <span class="st">""</span>,</span>
<span>  lwd <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  y.ticks <span class="op">=</span> <span class="fl">3</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ExpressionSet</dt>
<dd><p>a standard PhyloExpressionSet or DivergenceExpressionSet object.</p></dd>


<dt>measure</dt>
<dd><p>type of transcriptome index that shall be computed. E.g.</p><ul><li><p><code>measure = "TAI"</code> (Transcriptome Age Index)</p></li>
<li><p><code>measure = "TDI"</code> (Transcriptome Divergence Index)</p></li>
<li><p><code>measure = "TPI"</code> (Transcriptome Polymorphism Index)</p></li>
</ul></dd>


<dt>TestStatistic</dt>
<dd><p>a string defining the type of test statistics to be used to quantify the statistical significance the present phylotranscriptomics pattern.
Possible values can be:</p><ul><li><p><code>TestStatistic</code> = <code>"FlatLineTest"</code> : Statistical test for the deviation from a flat line</p></li>
<li><p><code>TestStatistic</code> = <code>"ReductiveHourglassTest"</code> : Statistical test for the existence of a hourglass shape (high-low-high pattern)</p></li>
<li><p><code>TestStatistic</code> = <code>"ReverseHourglassTest"</code> : Statistical test for the existence of a reverse hourglass pattern (low-high-low pattern)</p></li>
<li><p><code>TestStatistic</code> = <code>"EarlyConservationTest"</code> : Statistical test for the existence of a early conservation pattern (low-high-high pattern)</p></li>
<li><p><code>TestStatistic</code> = <code>"LateConservationTest"</code> : Statistical test for the existence of a late conservation pattern (high-high-low pattern)</p></li>
</ul></dd>


<dt>transforms</dt>
<dd><p>a character vector of any valid function that transforms gene expression levels. Available options are:</p><ul><li><p><code>transforms = "none"</code> : No transformation (absolute expression)</p></li>
<li><p><code>transforms = "log2"</code> : Computes binary (i.e., base 2) logarithms, \(log_{2}{X}\).</p></li>
<li><p><code>transforms = "log"</code> : Computes natural logarithms, \(log_{e}{X}\).</p></li>
<li><p><code>transforms = "log10"</code> : Computes common logarithms (i.e., base 10), \(log_{10}{X}\).</p></li>
<li><p><code>transforms = "sqrt"</code> : Computes the (principle) square root, \(\sqrt{X}\).</p></li>
<li><p><code>transforms = "vst"</code> :  (Quickly) estimates dispersion trend and applies a variance stabilizing transformation (please make sure that the <span class="pkg">DESeq2</span> package is installed).</p></li>
<li><p><code>transforms = "rlog"</code> : (Robustly) estimates dispersion trend and applies a variance stabilizing transformation (please make sure that the <span class="pkg">DESeq2</span> package is installed).</p></li>
<li><p><code>transforms = "rank"</code> : Ranks genes from lowest to highest based on their expression levels, at each condition (e.g., developmental stage). The gene's expression value is replaced by its sample rank or average ranks in case of ties.</p></li>
<li><p><code>transforms = "squared"</code>: Computes the square, \(X^2\).</p></li>
</ul></dd>


<dt>modules</dt>
<dd><p>a list storing three elements for the <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>, <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>, <code><a href="LateConservationTest.html">LateConservationTest</a></code>,
or <code><a href="ReverseHourglassTest.html">ReverseHourglassTest</a></code>: early, mid, and late.
Each element expects a numeric vector specifying the developmental stages
or experiments that correspond to each module. For example:</p><ul><li><p><code>module</code> = <code>list(early = 1:2, mid = 3:5, late = 6:7)</code> divides a dataset storing seven developmental stages into 3 modules.</p></li>
</ul></dd>


<dt>permutations</dt>
<dd><p>a numeric value specifying the number of permutations to be performed for the <code><a href="FlatLineTest.html">FlatLineTest</a></code>, <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>, <code><a href="LateConservationTest.html">LateConservationTest</a></code>, <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code> or <code><a href="ReverseHourglassTest.html">ReverseHourglassTest</a></code>.</p></dd>


<dt>pseudocount</dt>
<dd><p>any valid number to add to the expression matrix prior to log transformations.</p></dd>


<dt>p.value</dt>
<dd><p>a boolean value specifying whether the p-value of the test statistic shall be printed within the plot area.</p></dd>


<dt>shaded.area</dt>
<dd><p>a boolean value specifying whether a shaded area shall
be drawn for the developmental stages defined to be the presumptive phylotypic period.</p></dd>


<dt>xlab</dt>
<dd><p>label of x-axis.</p></dd>


<dt>ylab</dt>
<dd><p>label of y-axis.</p></dd>


<dt>main</dt>
<dd><p>figure title.</p></dd>


<dt>lwd</dt>
<dd><p>line width.</p></dd>


<dt>alpha</dt>
<dd><p>transparency of the shaded area (between [0,1]). Default is <code>alpha = 0.1</code>.</p></dd>


<dt>y.ticks</dt>
<dd><p>number of ticks on the y-axis. Default is <code>ticks = 3</code>.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a ggplot object containing the following information
for each visualised transcriptome indices:
<code>p.value</code> : the p-value quantifying the statistical significance (depending on the chosen test) of the given phylotranscriptomics pattern.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Visualisation and assessment for the stability of data transforms on the permutation test of choice.
For details, please consult the main function <code><a href="PlotSignature.html">PlotSignature</a></code>,
as well as <code><a href="tf.html">tf</a></code>, <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>,
<code><a href="FlatLineTest.html">FlatLineTest</a></code>, <code><a href="ReverseHourglassTest.html">ReverseHourglassTest</a></code>,
<code><a href="LateConservationTest.html">LateConservationTest</a></code> or <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>.</p>
<p>In most cases, users can replace <code><a href="PlotSignature.html">PlotSignature</a></code> simply with
<code>PlotSignatureTransformed</code> to obtain the multi-panel plot with different
transformations to visualise the stability of the pattern observed with <code><a href="PlotSignature.html">PlotSignature</a></code>.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Lotharukpong JS et al. (2023) bioRxiv</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="PlotSignature.html">PlotSignature</a></code>,<code><a href="tfStability.html">tfStability</a></code>,
<code><a href="FlatLineTest.html">FlatLineTest</a></code>, <code><a href="ReverseHourglassTest.html">ReverseHourglassTest</a></code>,
<code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>, <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>,
<code><a href="LateConservationTest.html">LateConservationTest</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Jaruwatana Sodai Lotharukpong</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Flat line test</span></span></span>
<span class="r-in"><span><span class="fu">PlotSignatureTransformed</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>                    TestStatistic <span class="op">=</span> <span class="st">"FlatLineTest"</span>,</span></span>
<span class="r-in"><span>                    transforms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"log2"</span>, <span class="st">"sqrt"</span>, <span class="st">"vst"</span>, <span class="st">"rank"</span>, <span class="st">"squared"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                    modules <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>early <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, mid <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">5</span>, late <span class="op">=</span> <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Reductive hourglass test</span></span></span>
<span class="r-in"><span><span class="fu">PlotSignatureTransformed</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>                     TestStatistic <span class="op">=</span> <span class="st">"ReductiveHourglassTest"</span>,</span></span>
<span class="r-in"><span>                     transforms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"log2"</span>, <span class="st">"sqrt"</span>, <span class="st">"rank"</span>, <span class="st">"squared"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                     modules <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>early <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, mid <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">5</span>, late <span class="op">=</span> <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">PlotSignatureTransformed</span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="va">PhyloExpressionSetExample</span>,</span></span>
<span class="r-in"><span>                     TestStatistic <span class="op">=</span> <span class="st">"ReductiveHourglassTest"</span>,</span></span>
<span class="r-in"><span>                     transforms <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"none"</span>, <span class="st">"log2"</span>, <span class="st">"sqrt"</span>, <span class="st">"vst"</span>, <span class="st">"rank"</span>, <span class="st">"squared"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                     modules <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>early <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">2</span>, mid <span class="op">=</span> <span class="fl">3</span><span class="op">:</span><span class="fl">5</span>, late <span class="op">=</span> <span class="fl">6</span><span class="op">:</span><span class="fl">7</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Kristian K Ullrich.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

