<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Plot evolutionary signatures across transcriptomes — PlotSignature • myTAI</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Plot evolutionary signatures across transcriptomes — PlotSignature"><meta property="og:description" content="Main function to visualize transcriptome indices."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI package</a>
    </li>
    <li>
      <a href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/myTAI/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Plot evolutionary signatures across transcriptomes</h1>
    <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/PlotSignature.R" class="external-link"><code>R/PlotSignature.R</code></a></small>
    <div class="hidden name"><code>PlotSignature.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Main function to visualize transcriptome indices.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">PlotSignature</span><span class="op">(</span></span>
<span>  <span class="va">ExpressionSet</span>,</span>
<span>  measure <span class="op">=</span> <span class="st">"TAI"</span>,</span>
<span>  TestStatistic <span class="op">=</span> <span class="st">"FlatLineTest"</span>,</span>
<span>  modules <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  permutations <span class="op">=</span> <span class="fl">1000</span>,</span>
<span>  lillie.test <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  p.value <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  shaded.area <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  custom.perm.matrix <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  xlab <span class="op">=</span> <span class="st">"Ontogeny"</span>,</span>
<span>  ylab <span class="op">=</span> <span class="st">"Transcriptome Index"</span>,</span>
<span>  main <span class="op">=</span> <span class="st">""</span>,</span>
<span>  lwd <span class="op">=</span> <span class="fl">4</span>,</span>
<span>  alpha <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>  y.ticks <span class="op">=</span> <span class="fl">10</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ExpressionSet</dt>
<dd><p>a standard PhyloExpressionSet, DivergenceExpressionSet or PolymorphismsExpressionSet object.</p></dd>


<dt>measure</dt>
<dd><p>type of transcriptome index that shall be computed. E.g.</p><ul><li><p><code>measure = "TAI"</code> (Transcriptome Age Index)</p></li>
<li><p><code>measure = "TDI"</code> (Transcriptome Divergence Index)</p></li>
<li><p><code>measure = "TPI"</code> (Transcriptome Polymorphism Index)</p></li>
</ul></dd>


<dt>TestStatistic</dt>
<dd><p>a string defining the type of test statistics to be used to quantify the statistical significance the present phylotranscriptomics pattern.
Possible values can be:</p><ul><li><p><code>TestStatistic</code> = <code>"FlatLineTest"</code> : Statistical test for the deviation from a flat line</p></li>
<li><p><code>TestStatistic</code> = <code>"ReductiveHourglassTest"</code> : Statistical test for the existence of a hourglass shape (high-low-high pattern)</p></li>
<li><p><code>TestStatistic</code> = <code>"EarlyConservationTest"</code> : Statistical test for the existence of a early conservation pattern (low-high-high pattern)</p></li>
<li><p><code>TestStatistic</code> = <code>"LateConservationTest"</code> : Statistical test for the existence of a late conservation pattern (high-high-low pattern)</p></li>
<li><p><code>TestStatistic</code> = <code>"ReverseHourglassTest"</code> : Statistical test for the existence of a reverse hourglass pattern (low-high-low pattern)</p></li>
</ul></dd>


<dt>modules</dt>
<dd><p>a list storing three elements for the <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>, <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>, <code><a href="LateConservationTest.html">LateConservationTest</a></code>, 
or <code><a href="ReverseHourglassTest.html">ReverseHourglassTest</a></code>: early, mid, and late. 
Each element expects a numeric vector specifying the developmental stages 
or experiments that correspond to each module. For example:</p><ul><li><p><code>module</code> = <code>list(early = 1:2, mid = 3:5, late = 6:7)</code> divides a dataset storing seven developmental stages into 3 modules.</p></li>
</ul></dd>


<dt>permutations</dt>
<dd><p>a numeric value specifying the number of permutations to be performed for the <code><a href="FlatLineTest.html">FlatLineTest</a></code>, <code><a href="EarlyConservationTest.html">EarlyConservationTest</a></code>, <code><a href="LateConservationTest.html">LateConservationTest</a></code>, <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code> or <code><a href="ReverseHourglassTest.html">ReverseHourglassTest</a></code>.</p></dd>


<dt>lillie.test</dt>
<dd><p>a boolean value specifying whether the Lilliefors Kolmogorov-Smirnov Test shall be performed.</p></dd>


<dt>p.value</dt>
<dd><p>a boolean value specifying whether the p-value of the test statistic shall be printed as a subtitle.</p></dd>


<dt>shaded.area</dt>
<dd><p>a boolean value specifying whether a shaded area shall 
be drawn for the developmental stages defined to be the presumptive phylotypic period.</p></dd>


<dt>custom.perm.matrix</dt>
<dd><p>a custom <code><a href="bootMatrix.html">bootMatrix</a></code> (permutation matrix) to perform the underlying test statistic visualized by <code>PlotSignature</code>. Default is <code>custom.perm.matrix = NULL</code>.</p></dd>


<dt>xlab</dt>
<dd><p>label of x-axis.</p></dd>


<dt>ylab</dt>
<dd><p>label of y-axis.</p></dd>


<dt>main</dt>
<dd><p>figure title.</p></dd>


<dt>lwd</dt>
<dd><p>line width.</p></dd>


<dt>alpha</dt>
<dd><p>transparency of the shaded area (between [0,1]). Default is <code>alpha = 0.1</code>.</p></dd>


<dt>y.ticks</dt>
<dd><p>number of ticks on the y-axis. Default is <code>ticks = 10</code>.</p></dd>

</dl></div>
    <div id="details">
    <h2>Details</h2>
    <p>This function substitutes the functionality of the <code><a href="PlotPattern.html">PlotPattern</a></code> function
and is based on ggplot2 insead of base R graphics.</p>
<p>The following transcriptome indices can be computed and visualized with this function:</p><ul><li><p>Transcriptome Age Index (<code><a href="TAI.html">TAI</a></code>)</p></li>
<li><p>Transcriptome Divergence Index (<code><a href="TDI.html">TDI</a></code>)</p></li>
<li><p>Transcriptome Polymorphism Index (<code><a href="TPI.html">TPI</a></code>)</p></li>
</ul></div>
    <div id="author">
    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># plot TAI pattern and perform flat line test</span></span></span>
<span class="r-in"><span><span class="fu">PlotSignature</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, </span></span>
<span class="r-in"><span>              measure       <span class="op">=</span> <span class="st">"TAI"</span>, </span></span>
<span class="r-in"><span>              permutations  <span class="op">=</span> <span class="fl">100</span>,</span></span>
<span class="r-in"><span>              TestStatistic <span class="op">=</span> <span class="st">"FlatLineTest"</span>,</span></span>
<span class="r-in"><span>              ylab <span class="op">=</span> <span class="st">"Transcriptome Age Index"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Plot signature: ' TAI ' and test statistic: ' FlatLineTest ' running  100  permutations.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> We recommended using at least 20000 permutations to achieve a sufficient permutation test.</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Significance status of signature:  significant.</span>
<span class="r-err co"><span class="r-pr">#&gt;</span> <span class="error">Error in TestStatistic == FlatLineTest:</span> comparison (==) is possible only for atomic and list types</span>
<span class="r-in"><span>              </span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Kristian K Ullrich.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

