<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Transform Gene Expression Levels — tf • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Transform Gene Expression Levels — tf"><meta property="og:description" content="This function transforms the gene expression set stored in an input 
PhloExpressionSet or DivergenceExpressionSet object and returns a 
PhloExpressionSet or DivergenceExpressionSet object with transformed expression levels. 
The resulting transformed PhloExpressionSet or DivergenceExpressionSet 
object can then be used for subsequent analyses based on transformed expression levels."><meta property="og:image" content="/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">myTAI</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.0.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Enrichment.html">Enrichment Analyses</a>
    </li>
    <li>
      <a href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a>
    </li>
    <li>
      <a href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a>
    </li>
    <li>
      <a href="../articles/Introduction.html">Introduction to the myTAI package</a>
    </li>
    <li>
      <a href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a>
    </li>
    <li>
      <a href="../articles/Taxonomy.html">Retrieving Taxonomic Information</a>
    </li>
  </ul></li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/drostlab/myTAI/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Transform Gene Expression Levels</h1>
    <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/tf.R" class="external-link"><code>R/tf.R</code></a></small>
    <div class="hidden name"><code>tf.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This function transforms the gene expression set stored in an input 
PhloExpressionSet or DivergenceExpressionSet object and returns a 
PhloExpressionSet or DivergenceExpressionSet object with transformed expression levels. 
The resulting transformed PhloExpressionSet or DivergenceExpressionSet 
object can then be used for subsequent analyses based on transformed expression levels.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">tf</span><span class="op">(</span><span class="va">ExpressionSet</span>, <span class="va">FUN</span>, pseudocount <span class="op">=</span> <span class="fl">0</span>, integerise <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>ExpressionSet</dt>
<dd><p>a standard PhloExpressionSet or DivergenceExpressionSet object.</p></dd>


<dt>FUN</dt>
<dd><p>any valid function that transforms gene expression levels.</p></dd>


<dt>pseudocount</dt>
<dd><p>a numeric value to be added to the expression matrix prior to transformation.</p></dd>


<dt>integerise</dt>
<dd><p>a boolean specifying whether the expression data should be rounded to the nearest integer.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>a standard PhloExpressionSet or DivergenceExpressionSet object storing transformed gene expression levels.</p>
    </div>
    <div id="details">
    <h2>Details</h2>
    <p>Motivated by the dicussion raised by Piasecka et al., 2013, the influence of
gene expression transformation on the global phylotranscriptomics pattern does not seem negligible.
Hence, different transformations can result in qualitatively different <code><a href="TAI.html">TAI</a></code> or <code><a href="TDI.html">TDI</a></code>
patterns.</p>
<p>Initially, the <code><a href="TAI.html">TAI</a></code> and <code><a href="TDI.html">TDI</a></code> formulas were defined for absolute expression levels.
So using the initial <code><a href="TAI.html">TAI</a></code> and <code><a href="TDI.html">TDI</a></code> formulas with transformed expression levels
might turn out in qualitatively different patterns when compared with non-transformed expression levels,
but might also belong to a different class of models, since different valid expression level transformation functions result in different patterns.</p>
<p>The purpose of this function is to allow the user to study the qualitative impact of different transformation functions on 
the global <code><a href="TAI.html">TAI</a></code> and <code><a href="TDI.html">TDI</a></code> pattern, or on any subsequent phylotranscriptomics analysis.</p>
<p>The examples using the <em>PhyloExpressionSetExample</em> data set show that using common gene expression 
transformation functions: <code><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></code> (Quackenbush, 2001 and 2002), <code><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></code> (Yeung et al., 2001), 
<code><a href="https://rdrr.io/pkg/MASS/man/boxcox.html" class="external-link">boxcox</a></code>, or <em>inverse hyperbolic sine transformation</em>, each transformation results 
in qualitatively different patterns. 
Nevertheless, for each resulting pattern the statistical significance can be tested 
using either the <code><a href="FlatLineTest.html">FlatLineTest</a></code> or <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code> (Drost et al., 2014) 
to quantify the significance of interest.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    <p>Piasecka B, Lichocki P, Moretti S, et al. (2013) The hourglass and the early conservation models--co-existing patterns of developmental constraints in vertebrates. PLoS Genet. 9(4): e1003476.</p>
<p>Quint M., Drost H.G., Gabel A., Ullrich K.K., Boenn M., Grosse I. (2012) A transcriptomic hourglass in plant embryogenesis. Nature 490: 98-101.</p>
<p>Domazet-Loso T., Tautz D. (2010) A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns. Nature 468: 815-8.</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012</p>
<p>K.Y. Yeung et al.: Model-based clustering and data transformations for gene expression data. Bioinformatics 2001, 17:977-987</p>
<p>K.Y. Yeung et al.:  Supplement to Model-based clustering and data transformations for gene expression data - Data Transformations and the Gaussian mixture assumption. Bioinformatics 2001, 17:977-987</p>
<p>P.A.C. Hoen et al.: Deep sequencing-based expression analysis shows major advances in robustness, resolution and inter-lab portability over five microarray platforms. Nucleic Acids Research 2008, Vol. 36, No. 21</p>
<p>H.H. Thygesen et al.: Comparing transformation methods for DNA microarray data. BMC Bioinformatics 2004, 5:77</p>
<p>John Quackenbush: Microarray data normalization and transformation. Nature Genetics 2002, 32:496-501</p>
<p>John Quackenbush: Computational Analysis of Microarray Data. Nature Reviews 2001, 2:418-427</p>
<p>R. Nadon and J. Shoemaker: Statistical issues with microarrays: processing and analysis. TRENDS in Genetics 2002, Vol. 18 No. 5:265-271</p>
<p>B.P. Durbin et al.: A variance-stabilizing transformation for gene-expression microarray data. Bioinformatics 2002, 18:S105-S110</p>
<p>J. M. Bland et al.: Transforming data. BMJ 1996, 312:770</p>
<p>John B. Burbidge, Lonnie Magee and A. Leslie Robb (1988) Alternative Transformations to Handle Extreme Values of the Dependent Variable. Journal of the American Statistical Association, 83(401): 123-127.</p>
<p>G. E. P. Box and D. R. Cox (1964) An Analysis of Transformations. Journal of the Royal Statistical Society. Series B (Methodological), 26(2): 211-252.</p>
    </div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index"><p><code><a href="TAI.html">TAI</a></code>, <code><a href="TDI.html">TDI</a></code>, <code><a href="FlatLineTest.html">FlatLineTest</a></code>, <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>,
<code><a href="tfStability.html">tfStability</a></code></p></div>
    </div>
    <div id="author">
    <h2>Author</h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># a simple example is to transform the gene expression levels</span></span></span>
<span class="r-in"><span><span class="co"># of a given PhyloExpressionSet using a sqrt or log2 transformation</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PES.sqrt</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">sqrt</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PES.log2</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">log2</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># in case a given PhyloExpressionSet already stores gene expression levels</span></span></span>
<span class="r-in"><span><span class="co"># that are log2 transformed and need to be re-transformed to absolute</span></span></span>
<span class="r-in"><span><span class="co"># expression levels, to perform subsequent phylotranscriptomics analyses </span></span></span>
<span class="r-in"><span><span class="co"># (that are defined for absolute expression levels), one can re-transform</span></span></span>
<span class="r-in"><span><span class="co"># a PhyloExpressionSet like this:</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PES.absolute</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PES.log2</span> , <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fl">2</span><span class="op">^</span><span class="va">x</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># which should be the same as PhyloExpressionSetExample :</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Phylostratum      GeneID     Zygote   Quadrant   Globular      Heart</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1 at1g01040.2  2173.6352  1911.2001  1152.5553  1291.4224</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1 at1g01050.1  1501.0141  1817.3086  1665.3089  1564.7612</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            1 at1g01070.1  1212.7927  1233.0023   939.2000   929.6195</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            1 at1g01080.2  1016.9203   936.3837  1181.3381  1329.4734</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            1 at1g01090.1 11424.5667 16778.1685 34366.6493 39775.6405</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6            1 at1g01120.1   844.0414   787.5929   859.6267   931.6180</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      Torpedo       Bent    Mature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  1000.2529   962.9772 1696.4274</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  1496.3207  1114.6435 1071.6555</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   864.2180   877.2060  894.8189</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  1392.6429  1287.9746  861.2605</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 56231.5689 66980.3673 7772.5617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   942.8453   870.2625  792.7542</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">PES.absolute</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Phylostratum      GeneID     Zygote   Quadrant   Globular      Heart</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1 at1g01040.2  2173.6352  1911.2001  1152.5553  1291.4224</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1 at1g01050.1  1501.0141  1817.3086  1665.3089  1564.7612</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            1 at1g01070.1  1212.7927  1233.0023   939.2000   929.6195</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            1 at1g01080.2  1016.9203   936.3837  1181.3381  1329.4734</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            1 at1g01090.1 11424.5667 16778.1685 34366.6493 39775.6405</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6            1 at1g01120.1   844.0414   787.5929   859.6267   931.6180</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      Torpedo       Bent    Mature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  1000.2529   962.9772 1696.4274</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  1496.3207  1114.6435 1071.6555</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   864.2180   877.2060  894.8189</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  1392.6429  1287.9746  861.2605</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 56231.5689 66980.3673 7772.5617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   942.8453   870.2625  792.7542</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># plotting the TAI using log2 transformed expression levels</span></span></span>
<span class="r-in"><span><span class="co"># and performing the Flat Line Test to obtain the p-value</span></span></span>
<span class="r-in"><span><span class="fu"><a href="PlotPattern.html">PlotPattern</a></span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">log2</span><span class="op">)</span>, </span></span>
<span class="r-in"><span>            type          <span class="op">=</span> <span class="st">"l"</span>, </span></span>
<span class="r-in"><span>            lwd           <span class="op">=</span> <span class="fl">5</span>, </span></span>
<span class="r-in"><span>            TestStatistic <span class="op">=</span> <span class="st">"FlatLineTest"</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Total runtime of your permutation test: 0.495  seconds.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -&gt; We recommended using at least 20000 permutations to achieve a sufficient permutation test.</span>
<span class="r-plt img"><img src="tf-1.png" alt="" width="700" height="433"></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># for square-transformation (not square-root), we recommend using </span></span></span>
<span class="r-in"><span><span class="co"># function(x) x*x rather than function(x) x^2, due to speed.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PES.absolute</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PES.sqrt</span> , <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="va">x</span><span class="op">*</span><span class="va">x</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># and the result should be the same as PhyloExpressionSetExample :</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Phylostratum      GeneID     Zygote   Quadrant   Globular      Heart</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1 at1g01040.2  2173.6352  1911.2001  1152.5553  1291.4224</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1 at1g01050.1  1501.0141  1817.3086  1665.3089  1564.7612</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            1 at1g01070.1  1212.7927  1233.0023   939.2000   929.6195</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            1 at1g01080.2  1016.9203   936.3837  1181.3381  1329.4734</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            1 at1g01090.1 11424.5667 16778.1685 34366.6493 39775.6405</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6            1 at1g01120.1   844.0414   787.5929   859.6267   931.6180</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      Torpedo       Bent    Mature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  1000.2529   962.9772 1696.4274</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  1496.3207  1114.6435 1071.6555</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   864.2180   877.2060  894.8189</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  1392.6429  1287.9746  861.2605</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 56231.5689 66980.3673 7772.5617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   942.8453   870.2625  792.7542</span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">PES.absolute</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   Phylostratum      GeneID     Zygote   Quadrant   Globular      Heart</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1            1 at1g01040.2  2173.6352  1911.2001  1152.5553  1291.4224</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2            1 at1g01050.1  1501.0141  1817.3086  1665.3089  1564.7612</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3            1 at1g01070.1  1212.7927  1233.0023   939.2000   929.6195</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4            1 at1g01080.2  1016.9203   936.3837  1181.3381  1329.4734</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5            1 at1g01090.1 11424.5667 16778.1685 34366.6493 39775.6405</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6            1 at1g01120.1   844.0414   787.5929   859.6267   931.6180</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>      Torpedo       Bent    Mature</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1  1000.2529   962.9772 1696.4274</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2  1496.3207  1114.6435 1071.6555</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3   864.2180   877.2060  894.8189</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4  1392.6429  1287.9746  861.2605</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5 56231.5689 66980.3673 7772.5617</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6   942.8453   870.2625  792.7542</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># in case the expression matrix contains 0s, a pseudocount can be added prior</span></span></span>
<span class="r-in"><span><span class="co"># to certain transformations, e.g. log2(x+1) where 1 is the pseudocount.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PhyloExpressionSetExample</span><span class="op">[</span><span class="fl">4</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">=</span> <span class="fl">0</span></span></span>
<span class="r-in"><span><span class="va">PES.log2</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">log2</span>, pseudocount <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># this should return -Inf at PES.log2[4,3] the issue here is that </span></span></span>
<span class="r-in"><span><span class="co"># -Inf cannot be used to compute the phylotranscriptomic profile.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PES.log2</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">log2</span>, pseudocount <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># log2 transformed expression levels can now be used in downstream analyses.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># to perform rank transformation</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PES.rank</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">2</span>, <span class="fu">base</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/r/base/rank.html" class="external-link">rank</a></span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># rlog and vst transformations are now also possible by loading the DESeq2 package</span></span></span>
<span class="r-in"><span><span class="co"># and transforming the data with the parameter integerise = TRUE.</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mikelove/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span> <span class="co"># make sure the DESeq2 version &gt;= 1.29.15 for rlog</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: S4Vectors</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: stats4</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: BiocGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘BiocGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:stats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     IQR, mad, sd, var, xtabs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Filter, Find, Map, Position, Reduce, anyDuplicated, aperm, append,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     as.data.frame, basename, cbind, colnames, dirname, do.call,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     duplicated, eval, evalq, get, grep, grepl, intersect, is.unsorted,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     lapply, mapply, match, mget, order, paste, pmax, pmax.int, pmin,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     pmin.int, rank, rbind, rownames, sapply, setdiff, sort, table,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     tapply, union, unique, unsplit, which.max, which.min</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘S4Vectors’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:base’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     I, expand.grid, unname</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: IRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomicRanges</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: GenomeInfoDb</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: SummarizedExperiment</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: MatrixGenerics</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: matrixStats</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘MatrixGenerics’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:matrixStats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colAlls, colAnyNAs, colAnys, colAvgsPerRowSet, colCollapse,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colCounts, colCummaxs, colCummins, colCumprods, colCumsums,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colDiffs, colIQRDiffs, colIQRs, colLogSumExps, colMadDiffs,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colMads, colMaxs, colMeans2, colMedians, colMins, colOrderStats,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colProds, colQuantiles, colRanges, colRanks, colSdDiffs, colSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colSums2, colTabulates, colVarDiffs, colVars, colWeightedMads,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedMeans, colWeightedMedians, colWeightedSds,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     colWeightedVars, rowAlls, rowAnyNAs, rowAnys, rowAvgsPerColSet,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCollapse, rowCounts, rowCummaxs, rowCummins, rowCumprods,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowCumsums, rowDiffs, rowIQRDiffs, rowIQRs, rowLogSumExps,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMadDiffs, rowMads, rowMaxs, rowMeans2, rowMedians, rowMins,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowOrderStats, rowProds, rowQuantiles, rowRanges, rowRanks,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowSdDiffs, rowSds, rowSums2, rowTabulates, rowVarDiffs, rowVars,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedMads, rowWeightedMeans, rowWeightedMedians,</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowWeightedSds, rowWeightedVars</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading required package: Biobase</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Welcome to Bioconductor</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     Vignettes contain introductory material; view with</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'browseVignettes()'. To cite Bioconductor, see</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     'citation("Biobase")', and for packages 'citation("pkgname")'.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attaching package: ‘Biobase’</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following object is masked from ‘package:MatrixGenerics’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     rowMedians</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> The following objects are masked from ‘package:matrixStats’:</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>     anyMissing, rowMedians</span>
<span class="r-in"><span><span class="va">PES.vst</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">vst</span>, integerise <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> converting counts to integer mode</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> -- note: fitType='parametric', but the dispersion trend was not well captured by the</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    function: y = a/x + b, and a local regression fit was automatically substituted.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>    specify fitType='local' or 'mean' to avoid this message next time.</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Hajk-Georg Drost, Kristian K Ullrich, Jaruwatana Sodai Lotharukpong, Nikola Kalábová.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

