<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Transform Gene Expression Levels — tf • myTAI</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Transform Gene Expression Levels — tf"><meta name="description" content="This function transforms the gene expression set stored in an input
PhloExpressionSet or DivergenceExpressionSet object and returns a
PhloExpressionSet or DivergenceExpressionSet object with transformed expression levels.
The resulting transformed PhloExpressionSet or DivergenceExpressionSet
object can then be used for subsequent analyses based on transformed expression levels."><meta property="og:description" content="This function transforms the gene expression set stored in an input
PhloExpressionSet or DivergenceExpressionSet object and returns a
PhloExpressionSet or DivergenceExpressionSet object with transformed expression levels.
The resulting transformed PhloExpressionSet or DivergenceExpressionSet
object can then be used for subsequent analyses based on transformed expression levels."><meta property="og:image" content="https://drostlab.github.io/myTAI/logo.png"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">myTAI</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Advanced.html">Advanced Phylotranscriptomics Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Enrichment.html">Enrichment Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Expression.html">Gene Expression Analysis with `myTAI`</a></li>
    <li><a class="dropdown-item" href="../articles/Intermediate.html">Intermediate Phylotranscriptomics Analyses</a></li>
    <li><a class="dropdown-item" href="../articles/Introduction.html">Introduction to the myTAI package</a></li>
    <li><a class="dropdown-item" href="../articles/Phylostratigraphy.html">Discussing Genomic Phylostratigraphy</a></li>
  </ul></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/drostlab/myTAI/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Transform Gene Expression Levels</h1>
      <small class="dont-index">Source: <a href="https://github.com/drostlab/myTAI/blob/HEAD/R/tf.R" class="external-link"><code>R/tf.R</code></a></small>
      <div class="d-none name"><code>tf.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function transforms the gene expression set stored in an input
PhloExpressionSet or DivergenceExpressionSet object and returns a
PhloExpressionSet or DivergenceExpressionSet object with transformed expression levels.
The resulting transformed PhloExpressionSet or DivergenceExpressionSet
object can then be used for subsequent analyses based on transformed expression levels.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">tf</span><span class="op">(</span><span class="va">ExpressionSet</span>, <span class="va">FUN</span>, pseudocount <span class="op">=</span> <span class="fl">0</span>, integerise <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-expressionset">ExpressionSet<a class="anchor" aria-label="anchor" href="#arg-expressionset"></a></dt>
<dd><p>a standard PhloExpressionSet or DivergenceExpressionSet object.</p></dd>


<dt id="arg-fun">FUN<a class="anchor" aria-label="anchor" href="#arg-fun"></a></dt>
<dd><p>any valid function that transforms gene expression levels.</p></dd>


<dt id="arg-pseudocount">pseudocount<a class="anchor" aria-label="anchor" href="#arg-pseudocount"></a></dt>
<dd><p>a numeric value to be added to the expression matrix prior to transformation.</p></dd>


<dt id="arg-integerise">integerise<a class="anchor" aria-label="anchor" href="#arg-integerise"></a></dt>
<dd><p>a boolean specifying whether the expression data should be rounded to the nearest integer.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>a standard PhloExpressionSet or DivergenceExpressionSet object storing transformed gene expression levels.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>Motivated by the dicussion raised by Piasecka et al., 2013, the influence of
gene expression transformation on the global phylotranscriptomics pattern does not seem negligible.
Hence, different transformations can result in qualitatively different <code><a href="TAI.html">TAI</a></code> or <code><a href="TDI.html">TDI</a></code>
patterns.</p>
<p>Initially, the <code><a href="TAI.html">TAI</a></code> and <code><a href="TDI.html">TDI</a></code> formulas were defined for absolute expression levels.
So using the initial <code><a href="TAI.html">TAI</a></code> and <code><a href="TDI.html">TDI</a></code> formulas with transformed expression levels
might turn out in qualitatively different patterns when compared with non-transformed expression levels,
but might also belong to a different class of models, since different valid expression level transformation functions result in different patterns.</p>
<p>The purpose of this function is to allow the user to study the qualitative impact of different transformation functions on
the global <code><a href="TAI.html">TAI</a></code> and <code><a href="TDI.html">TDI</a></code> pattern, or on any subsequent phylotranscriptomics analysis.</p>
<p>The examples using the <em>PhyloExpressionSetExample</em> data set show that using common gene expression
transformation functions: <code><a href="https://rdrr.io/r/base/Log.html" class="external-link">log2</a></code> (Quackenbush, 2001 and 2002), <code><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">sqrt</a></code> (Yeung et al., 2001),
<code><a href="https://rdrr.io/pkg/MASS/man/boxcox.html" class="external-link">boxcox</a></code>, or <em>inverse hyperbolic sine transformation</em>, each transformation results
in qualitatively different patterns.
Nevertheless, for each resulting pattern the statistical significance can be tested
using either the <code><a href="FlatLineTest.html">FlatLineTest</a></code> or <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code> (Drost et al., 2014)
to quantify the significance of interest.</p>
    </div>
    <div class="section level2">
    <h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a></h2>
    <p>Piasecka B, Lichocki P, Moretti S, et al. (2013) The hourglass and the early conservation models–co-existing patterns of developmental constraints in vertebrates. PLoS Genet. 9(4): e1003476.</p>
<p>Quint M., Drost H.G., Gabel A., Ullrich K.K., Boenn M., Grosse I. (2012) A transcriptomic hourglass in plant embryogenesis. Nature 490: 98-101.</p>
<p>Domazet-Loso T., Tautz D. (2010) A phylogenetically based transcriptome age index mirrors ontogenetic divergence patterns. Nature 468: 815-8.</p>
<p>Drost HG et al. (2015) Mol Biol Evol. 32 (5): 1221-1231 doi:10.1093/molbev/msv012</p>
<p>K.Y. Yeung et al.: Model-based clustering and data transformations for gene expression data. Bioinformatics 2001, 17:977-987</p>
<p>K.Y. Yeung et al.:  Supplement to Model-based clustering and data transformations for gene expression data - Data Transformations and the Gaussian mixture assumption. Bioinformatics 2001, 17:977-987</p>
<p>P.A.C. Hoen et al.: Deep sequencing-based expression analysis shows major advances in robustness, resolution and inter-lab portability over five microarray platforms. Nucleic Acids Research 2008, Vol. 36, No. 21</p>
<p>H.H. Thygesen et al.: Comparing transformation methods for DNA microarray data. BMC Bioinformatics 2004, 5:77</p>
<p>John Quackenbush: Microarray data normalization and transformation. Nature Genetics 2002, 32:496-501</p>
<p>John Quackenbush: Computational Analysis of Microarray Data. Nature Reviews 2001, 2:418-427</p>
<p>R. Nadon and J. Shoemaker: Statistical issues with microarrays: processing and analysis. TRENDS in Genetics 2002, Vol. 18 No. 5:265-271</p>
<p>B.P. Durbin et al.: A variance-stabilizing transformation for gene-expression microarray data. Bioinformatics 2002, 18:S105-S110</p>
<p>J. M. Bland et al.: Transforming data. BMJ 1996, 312:770</p>
<p>John B. Burbidge, Lonnie Magee and A. Leslie Robb (1988) Alternative Transformations to Handle Extreme Values of the Dependent Variable. Journal of the American Statistical Association, 83(401): 123-127.</p>
<p>G. E. P. Box and D. R. Cox (1964) An Analysis of Transformations. Journal of the Royal Statistical Society. Series B (Methodological), 26(2): 211-252.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="TAI.html">TAI</a></code>, <code><a href="TDI.html">TDI</a></code>, <code><a href="FlatLineTest.html">FlatLineTest</a></code>, <code><a href="ReductiveHourglassTest.html">ReductiveHourglassTest</a></code>,
<code><a href="tfStability.html">tfStability</a></code></p></div>
    </div>
    <div class="section level2">
    <h2 id="author">Author<a class="anchor" aria-label="anchor" href="#author"></a></h2>
    <p>Hajk-Georg Drost</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># a simple example is to transform the gene expression levels</span></span></span>
<span class="r-in"><span><span class="co"># of a given PhyloExpressionSet using a sqrt or log2 transformation</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PES.sqrt</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">sqrt</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PES.log2</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">log2</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># plotting the TAI using log2 transformed expression levels</span></span></span>
<span class="r-in"><span><span class="co"># and performing the Flat Line Test to obtain the p-value</span></span></span>
<span class="r-in"><span><span class="fu"><a href="PlotSignature.html">PlotSignature</a></span><span class="op">(</span>ExpressionSet <span class="op">=</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">log2</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>            permutations <span class="op">=</span> <span class="fl">1000</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># in case the expression matrix contains 0s, a pseudocount can be added prior</span></span></span>
<span class="r-in"><span><span class="co"># to certain transformations, e.g. log2(x+1) where 1 is the pseudocount.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PhyloExpressionSetExample</span><span class="op">[</span><span class="fl">4</span>,<span class="fl">3</span><span class="op">]</span> <span class="op">=</span> <span class="fl">0</span></span></span>
<span class="r-in"><span><span class="va">PES.log2</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">log2</span>, pseudocount <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># this should return -Inf at PES.log2[4,3] the issue here is that </span></span></span>
<span class="r-in"><span><span class="co"># -Inf cannot be used to compute the phylotranscriptomic profile.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PES.log2</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">log2</span>, pseudocount <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># log2 transformed expression levels can now be used in downstream analyses.</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># to perform rank transformation</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">PES.rank</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">x</span>, <span class="fl">2</span>, <span class="fu">base</span><span class="fu">::</span><span class="va"><a href="https://rdrr.io/r/base/rank.html" class="external-link">rank</a></span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># rlog and vst transformations are now also possible by loading the DESeq2 package</span></span></span>
<span class="r-in"><span><span class="co"># and transforming the data with the parameter integerise = TRUE.</span></span></span>
<span class="r-in"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/thelovelab/DESeq2" class="external-link">DESeq2</a></span><span class="op">)</span> <span class="co"># make sure the DESeq2 version &gt;= 1.29.15 for rlog</span></span></span>
<span class="r-in"><span><span class="va">PES.vst</span> <span class="op">&lt;-</span> <span class="fu">tf</span><span class="op">(</span><span class="va">PhyloExpressionSetExample</span>, <span class="va">vst</span>, integerise <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Hajk-Georg Drost, Jaruwatana Sodai Lotharukpong, Nikola Kalábová, Filipa Martins Costa, Kristian K Ullrich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

